---
title: "Mapped mcy reads to Teal URDB"
output: html_notebook
---

This analysis mapped Illumina short reads to the mcy gene cluster from several Microcystis genomes using BLAST. The reads that had alignments at 95% identity or greater with an alignment coverage of 80% or more of read length were extracted and aligned to Teal's URDB to screen for non-specific mapping. Focused on sample 49615, which showed that 200% of the Microcystis cells present had mcyD genes. Shouldn't be possible because all genomes currently available only have 1 copy of the mcy gene cluster in the genome.

See notes in /geomicro/data22/smitdere/Rerun_mcy_blasts/Mcy_BLAST_README.txt for the blasts and URDB alignments were done. This analysis was done on Vondamm.

Here, we are plotting and visualizing the output data.

Mcy gene cluster mapping
---

First, load the dataframes:  
```{r}
#Set working directory:
setwd("/Volumes/T7/Research/2014_Erie_Bloom/mcy_blasts/BLAST_new_QC/Mapping_troubleshooting")
#load required packages:
library(dplyr)
library(ggplot2)
library(patchwork)
library(tidyr)
library(zoo)
#Load the dataframes 
fwd_100_95_v_URDB_counts <- read.table("49615_fwd_mapped_mcy_100_95_vs_URDB.gene_counts.txt", sep='\t', header=FALSE)
colnames(fwd_100_95_v_URDB_counts) <- c("ClusterID", "FWD_Counts")

fwd_100_95_v_URDB_info <- read.table("49615_fwd_mapped_mcy_100_95_vs_URDB.info.txt", sep='\t', header=FALSE)
colnames(fwd_100_95_v_URDB_info) <- c("ClusterID", "Gene_name", "Gene_info", "Source_LCA", "Worst_case_LCA", 
                                      "count match at 95%", "LCA_100_pid", "LCA_95_pid", "contamination_flag", "KEGG", "PFAM",
                                      "COG", "GO", "Interpro", "EC", "Metal_binding", "location", "reaction", "metacyc_id",
                                      "rxn_direction", "reactants", "products", "protein_seq", "gene_seq")

rev_100_95_v_URDB_counts <- read.table("49615_rev_mapped_mcy_100_95_vs_URDB.gene_counts.txt", sep='\t', header=FALSE)
colnames(rev_100_95_v_URDB_counts) <- c("ClusterID", "REV_Counts")

rev_100_95_v_URDB_info <- read.table("49615_rev_mapped_mcy_100_95_vs_URDB.info.txt", sep='\t', header=FALSE)
colnames(rev_100_95_v_URDB_info) <- c("ClusterID", "Gene_name", "Gene_info", "Source_LCA", "Worst_case_LCA", 
                                      "count match at 95%", "LCA_100_pid", "LCA_95_pid", "contamination_flag", "KEGG", "PFAM",
                                      "COG", "GO", "Interpro", "EC", "Metal_binding", "location", "reaction", "metacyc_id",
                                      "rxn_direction", "reactants", "products", "protein_seq", "gene_seq")

fwd_95_90_v_URDB_counts <- read.table("49615_fwd_mapped_mcy_95_90_vs_URDB.gene_counts.txt", sep='\t', header=FALSE)
colnames(fwd_95_90_v_URDB_counts) <- c("ClusterID", "FWD_Counts")

fwd_95_90_v_URDB_info <- read.table("49615_fwd_mapped_mcy_95_90_vs_URDB.info.txt", sep='\t', header=FALSE)
colnames(fwd_95_90_v_URDB_info) <- c("ClusterID", "Gene_name", "Gene_info", "Source_LCA", "Worst_case_LCA", 
                                      "count match at 95%", "LCA_100_pid", "LCA_95_pid", "contamination_flag", "KEGG", "PFAM",
                                      "COG", "GO", "Interpro", "EC", "Metal_binding", "location", "reaction", "metacyc_id",
                                      "rxn_direction", "reactants", "products", "protein_seq", "gene_seq")

rev_95_90_v_URDB_counts <- read.table("49615_rev_mapped_mcy_95_90_vs_URDB.gene_counts.txt", sep='\t', header=FALSE)
colnames(rev_95_90_v_URDB_counts) <- c("ClusterID", "REV_Counts")

rev_95_90_v_URDB_info <- read.table("49615_rev_mapped_mcy_95_90_vs_URDB.info.txt", sep='\t', header=FALSE)
colnames(rev_95_90_v_URDB_info) <- c("ClusterID", "Gene_name", "Gene_info", "Source_LCA", "Worst_case_LCA", 
                                      "count match at 95%", "LCA_100_pid", "LCA_95_pid", "contamination_flag", "KEGG", "PFAM",
                                      "COG", "GO", "Interpro", "EC", "Metal_binding", "location", "reaction", "metacyc_id",
                                      "rxn_direction", "reactants", "products", "protein_seq", "gene_seq")

#Filter out unneeded columns in the info data frames:
#These are the columns that I want
keep <- c("ClusterID", "Gene_name", "LCA_100_pid", "LCA_95_pid", "products")
#Alter the data tables so that they only contain those columns:  
fwd_100_95_v_URDB_info <- fwd_100_95_v_URDB_info[ , colnames(fwd_100_95_v_URDB_info) %in% keep ]
rev_100_95_v_URDB_info <- rev_100_95_v_URDB_info[ , colnames(rev_100_95_v_URDB_info) %in% keep ]
fwd_95_90_v_URDB_info <- fwd_95_90_v_URDB_info[ , colnames(fwd_95_90_v_URDB_info) %in% keep ]
rev_95_90_v_URDB_info <- rev_95_90_v_URDB_info[ , colnames(rev_95_90_v_URDB_info) %in% keep ]

#Merge the count and info dataframes:
fwd_100_95_v_URDB_merged <- merge(fwd_100_95_v_URDB_counts, fwd_100_95_v_URDB_info, by="ClusterID", all=TRUE)
rev_100_95_v_URDB_merged <- merge(rev_100_95_v_URDB_counts, rev_100_95_v_URDB_info, by="ClusterID", all=TRUE)
fwd_95_90_v_URDB_merged <- merge(fwd_95_90_v_URDB_counts, fwd_95_90_v_URDB_info, by="ClusterID", all=TRUE)
rev_95_90_v_URDB_merged <- merge(rev_95_90_v_URDB_counts, rev_95_90_v_URDB_info, by="ClusterID", all=TRUE)

#Merge forward and reverse dataframes:
total_100_95_v_URDB_merged <- merge(fwd_100_95_v_URDB_merged, rev_100_95_v_URDB_merged, by=c("ClusterID", "Gene_name", "LCA_100_pid",
                                                                                             "LCA_95_pid", "products"), all=TRUE)
total_95_90_v_URDB_merged <- merge(fwd_95_90_v_URDB_merged, rev_95_90_v_URDB_merged, by=c("ClusterID", "Gene_name", "LCA_100_pid",
                                                                                             "LCA_95_pid", "products"), all=TRUE)

#Set NAs in the read count columns in the merged dataframes to 0:
total_100_95_v_URDB_merged$FWD_Counts[is.na(total_100_95_v_URDB_merged$FWD_Counts)] <- 0
total_100_95_v_URDB_merged$REV_Counts[is.na(total_100_95_v_URDB_merged$REV_Counts)] <- 0
total_95_90_v_URDB_merged$FWD_Counts[is.na(total_95_90_v_URDB_merged$FWD_Counts)] <- 0
total_95_90_v_URDB_merged$REV_Counts[is.na(total_95_90_v_URDB_merged$REV_Counts)] <- 0

#Sum the FWD and REV counts:
total_100_95_v_URDB_merged$Total_Counts <- total_100_95_v_URDB_merged$FWD_Counts + total_100_95_v_URDB_merged$REV_Counts
total_95_90_v_URDB_merged$Total_Counts <- total_95_90_v_URDB_merged$FWD_Counts + total_95_90_v_URDB_merged$REV_Counts
```

What proportion of the reads are mapping to mcy vs other genes/functions at 100-95 % nucleotide identity?  
```{r}
#Sum the data by gene_name:
total_100_95_summed_by_gene_name <- total_100_95_v_URDB_merged %>%
  group_by(Gene_name) %>%
  summarise(Total=sum(Total_Counts))

#Create a bar graph of the Total number of reads mapped to the different gene categories:
top25_100_95_gene_plot <- total_100_95_summed_by_gene_name %>%
                            arrange(desc(Total)) %>%
                              slice(1:25) %>%
                                ggplot(aes(x=reorder(Gene_name, -Total), y=Total)) +
                                  geom_bar(stat = "identity", position = position_dodge()) +
                                  theme_classic() +
                                  ggtitle("Top 25 functions") +
                                  theme(
                                    axis.line.x = element_line(size=0.1),
                                    axis.line.y = element_line(size=0.1),
                                    axis.title.x = element_text(size=12, margin = margin(t = 10, r = 0, b = 0, l = 0)), 
                                    axis.title.y = element_text(size=12, margin = margin(t = 0, r = 10, b = 0, l = 0)),
                                    panel.grid.major = element_blank(),
                                    panel.grid.minor = element_blank(),
                                    axis.text.x = element_text(size = 6, color = "black", angle = 45, vjust = 1, hjust = 1,
                                                               margin = margin(t = 10, r = 0, b = 0, l = 0)),
                                    axis.text.y = element_text(size = 10, color = "black", margin = margin(t = 0, r = 10, b = 0, l = 0)),
                                    axis.ticks.length = unit(-0.1, "cm"),
                                    axis.ticks = element_line(size = 0.1)
                                    ) +
                                  ylab("Total reads mapped") +
                                  xlab("") 


Second_chunk_100_95_gene_plot <- total_100_95_summed_by_gene_name %>%
                                    arrange(desc(Total)) %>%
                                      slice(26:50) %>%
                                        ggplot(aes(x=reorder(Gene_name, -Total), y=Total)) +
                                          geom_bar(stat = "identity", position = position_dodge()) +
                                          theme_classic() +
                                          ggtitle("Next 25 functions") +
                                          theme(
                                            axis.line.x = element_line(size=0.1),
                                            axis.line.y = element_line(size=0.1),
                                            axis.title.x = element_text(size=12, margin = margin(t = 10, r = 0, b = 0, l = 0)), 
                                            axis.title.y = element_text(size=12, margin = margin(t = 0, r = 10, b = 0, l = 0)),
                                            panel.grid.major = element_blank(),
                                            panel.grid.minor = element_blank(),
                                            axis.text.x = element_text(size = 6, color = "black", angle = 45, vjust = 1, hjust = 1,
                                                                       margin = margin(t = 10, r = 0, b = 0, l = 0)),
                                            axis.text.y = element_text(size = 10, color = "black", margin = margin(t = 0, r = 10, b = 0, l = 0)),
                                            axis.ticks.length = unit(-0.1, "cm"),
                                            axis.ticks = element_line(size = 0.1)
                                            ) +
                                          ylab("Total reads mapped") +
                                          xlab("") 

total_100_95_gene_plot <- top25_100_95_gene_plot + Second_chunk_100_95_gene_plot
total_100_95_gene_plot
ggsave("total_100_95_gene_plot.pdf", plot = total_100_95_gene_plot, width = 22, height = 18, units = "cm", dpi=320)
```
What proportion of the reads are mapping to Microcystis vs other organisms at 100-95% nucleotide identity?:
```{r}
#Sum the data by gene_name:
total_100_95_summed_by_LCA <- total_100_95_v_URDB_merged %>%
  group_by(LCA_100_pid) %>%
  summarise(Total=sum(Total_Counts))

#Divide the LCA taxonomy into different taxonomic ranks:
#Note that there is an unequal number of ranks listed for each row because Teal's pipeline only lists up to the lowest possible taxonomic rank So this command will generate errors about not enough fields expected. This is OK, the fields with missing data will simply be empty.
total_100_95_summed_by_LCA <- separate(total_100_95_summed_by_LCA, LCA_100_pid, c("Domain", "Phylum", "Class", "Order", "Family", "Genus", "Species"), sep=";", remove=TRUE)

#Transpose the table
total_100_95_summed_by_LCA <- t(total_100_95_summed_by_LCA)

#Fill the NAs in the taxonomy with the last non-NA value, the reorient to the correct position:
total_100_95_summed_by_LCA <- na.locf(total_100_95_summed_by_LCA)
total_100_95_summed_by_LCA <- as.data.frame(t(total_100_95_summed_by_LCA))
total_100_95_summed_by_LCA$Total <- as.numeric(total_100_95_summed_by_LCA$Total)

#Create a bar graph of the Total number of reads mapped to the different gene categories:
P100_95_LCA_plot <- total_100_95_summed_by_LCA %>%
                            arrange(desc(Total)) %>%
                                ggplot(aes(x=reorder(Genus, -Total), y=Total)) +
                                  geom_bar(stat = "identity", position = position_dodge()) +
                                  theme_classic() +
                                  theme(
                                    axis.line.x = element_line(size=0.1),
                                    axis.line.y = element_line(size=0.1),
                                    axis.title.x = element_text(size=12, margin = margin(t = 10, r = 0, b = 0, l = 0)), 
                                    axis.title.y = element_text(size=12, margin = margin(t = 0, r = 10, b = 0, l = 0)),
                                    panel.grid.major = element_blank(),
                                    panel.grid.minor = element_blank(),
                                    axis.text.x = element_text(size = 6, color = "black", angle = 45, vjust = 1, hjust = 1,
                                                               margin = margin(t = 10, r = 0, b = 0, l = 0)),
                                    axis.text.y = element_text(size = 10, color = "black", margin = margin(t = 0, r = 10, b = 0, l = 0)),
                                    axis.ticks.length = unit(-0.1, "cm"),
                                    axis.ticks = element_line(size = 0.1)
                                    ) +
                                  ylab("Total reads mapped") +
                                  xlab("Taxon") 

P100_95_LCA_plot
ggsave("P100_95_LCA_plot.pdf", plot = P100_95_LCA_plot, width = 22, height = 18, units = "cm", dpi=320)
```
What proportion of the reads are mapping to mcy vs other genes/functions at 95-90 % nucleotide identity?  
```{r}
#Sum the data by gene_name:
total_95_90_summed_by_gene_name <- total_95_90_v_URDB_merged %>%
  group_by(Gene_name) %>%
  summarise(Total=sum(Total_Counts))

#Create a bar graph of the Total number of reads mapped to the different gene categories:
top25_95_90_gene_plot <- total_95_90_summed_by_gene_name %>%
                            arrange(desc(Total)) %>%
                              slice(1:25) %>%
                                ggplot(aes(x=reorder(Gene_name, -Total), y=Total)) +
                                  geom_bar(stat = "identity", position = position_dodge()) +
                                  theme_classic() +
                                  ggtitle("Top 25 functions") +
                                  theme(
                                    axis.line.x = element_line(size=0.1),
                                    axis.line.y = element_line(size=0.1),
                                    axis.title.x = element_text(size=12, margin = margin(t = 10, r = 0, b = 0, l = 0)), 
                                    axis.title.y = element_text(size=12, margin = margin(t = 0, r = 10, b = 0, l = 0)),
                                    panel.grid.major = element_blank(),
                                    panel.grid.minor = element_blank(),
                                    axis.text.x = element_text(size = 6, color = "black", angle = 45, vjust = 1, hjust = 1,
                                                               margin = margin(t = 10, r = 0, b = 0, l = 0)),
                                    axis.text.y = element_text(size = 10, color = "black", margin = margin(t = 0, r = 10, b = 0, l = 0)),
                                    axis.ticks.length = unit(-0.1, "cm"),
                                    axis.ticks = element_line(size = 0.1)
                                    ) +
                                  ylab("Total reads mapped") +
                                  xlab("") 


Second_chunk_95_90_gene_plot <- total_95_90_summed_by_gene_name %>%
                                    arrange(desc(Total)) %>%
                                      slice(26:50) %>%
                                        ggplot(aes(x=reorder(Gene_name, -Total), y=Total)) +
                                          geom_bar(stat = "identity", position = position_dodge()) +
                                          theme_classic() +
                                          ggtitle("Next 25 functions") +
                                          theme(
                                            axis.line.x = element_line(size=0.1),
                                            axis.line.y = element_line(size=0.1),
                                            axis.title.x = element_text(size=12, margin = margin(t = 10, r = 0, b = 0, l = 0)), 
                                            axis.title.y = element_text(size=12, margin = margin(t = 0, r = 10, b = 0, l = 0)),
                                            panel.grid.major = element_blank(),
                                            panel.grid.minor = element_blank(),
                                            axis.text.x = element_text(size = 6, color = "black", angle = 45, vjust = 1, hjust = 1,
                                                                       margin = margin(t = 10, r = 0, b = 0, l = 0)),
                                            axis.text.y = element_text(size = 10, color = "black", margin = margin(t = 0, r = 10, b = 0, l = 0)),
                                            axis.ticks.length = unit(-0.1, "cm"),
                                            axis.ticks = element_line(size = 0.1)
                                            ) +
                                          ylab("Total reads mapped") +
                                          xlab("") 

total_95_90_gene_plot <- top25_95_90_gene_plot + Second_chunk_95_90_gene_plot
total_95_90_gene_plot
ggsave("total_95_90_gene_plot.pdf", plot = total_95_90_gene_plot, width = 22, height = 18, units = "cm", dpi=320)
```
What proportion of the reads are mapping to Microcystis vs other organisms at 95-90% nucleotide identity?:
```{r}
#Sum the data by LCA:
total_95_90_summed_by_LCA <- total_95_90_v_URDB_merged %>%
  group_by(LCA_100_pid) %>%
  summarise(Total=sum(Total_Counts))

#Divide the LCA taxonomy into different taxonomic ranks:
#Note that there is an unequal number of ranks listed for each row because Teal's pipeline only lists up to the lowest possible taxonomic rank So this command will generate errors about not enough fields expected. This is OK, the fields with missing data will simply be empty.
total_95_90_summed_by_LCA <- separate(total_95_90_summed_by_LCA, LCA_100_pid, c("Domain", "Phylum", "Class", "Order", "Family", "Genus", "Species"), sep=";", remove=TRUE)

#Transpose the table
total_95_90_summed_by_LCA <- t(total_95_90_summed_by_LCA)

#Fill the NAs in the taxonomy with the last non-NA value, the reorient to the correct position:
total_95_90_summed_by_LCA <- na.locf(total_95_90_summed_by_LCA)
total_95_90_summed_by_LCA <- as.data.frame(t(total_95_90_summed_by_LCA))
total_95_90_summed_by_LCA$Total <- as.numeric(total_95_90_summed_by_LCA$Total) #conver the counts back to numeric data

#Create a bar graph of the Total number of reads mapped to the different gene categories:
P95_90_LCA_plot <- total_95_90_summed_by_LCA %>%
                            arrange(desc(Total)) %>%
                                ggplot(aes(x=reorder(Genus, -Total), y=Total)) +
                                  geom_bar(stat = "identity", position = position_dodge()) +
                                  theme_classic() +
                                  theme(
                                    axis.line.x = element_line(size=0.1),
                                    axis.line.y = element_line(size=0.1),
                                    axis.title.x = element_text(size=12, margin = margin(t = 10, r = 0, b = 0, l = 0)), 
                                    axis.title.y = element_text(size=12, margin = margin(t = 0, r = 10, b = 0, l = 0)),
                                    panel.grid.major = element_blank(),
                                    panel.grid.minor = element_blank(),
                                    axis.text.x = element_text(size = 6, color = "black", angle = 45, vjust = 1, hjust = 1,
                                                               margin = margin(t = 10, r = 0, b = 0, l = 0)),
                                    axis.text.y = element_text(size = 10, color = "black", margin = margin(t = 0, r = 10, b = 0, l = 0)),
                                    axis.ticks.length = unit(-0.1, "cm"),
                                    axis.ticks = element_line(size = 0.1)
                                    ) +
                                  ylab("Total reads mapped") +
                                  xlab("Taxon") 

P95_90_LCA_plot
ggsave("P95_90_LCA_plot.pdf", plot = P95_90_LCA_plot, width = 22, height = 18, units = "cm", dpi=320)
```
Now let's see what the reads match to in Teal's URDB using muscato (DNA-DNA alignments):

Load the dataframes with the MUSCATO results:
```{r}
fwd_100_95_v_musc_URDB_counts <- read.table("49615_fwd_mapped_mcy_100_95_muscato_matches_genestats.txt", sep='\t', header=FALSE)
colnames(fwd_100_95_v_musc_URDB_counts) <- c("ClusterID", "FWD_Counts")

fwd_100_95_v_musc_URDB_info <- read.table("49615_fwd_mapped_mcy_100_95.muscato_info.txt", sep='\t', header=FALSE)
colnames(fwd_100_95_v_musc_URDB_info) <- c("ClusterID", "Gene_name", "Gene_info", "Source_LCA", "Worst_case_LCA", 
                                      "count match at 95%", "LCA_100_pid", "LCA_95_pid", "contamination_flag", "KEGG", "PFAM",
                                      "COG", "GO", "Interpro", "EC", "Metal_binding", "location", "reaction", "metacyc_id",
                                      "rxn_direction", "reactants", "products", "protein_seq", "gene_seq")

rev_100_95_v_musc_URDB_counts <- read.table("49615_rev_mapped_mcy_100_95_muscato_matches_genestats.txt", sep='\t', header=FALSE)
colnames(rev_100_95_v_musc_URDB_counts) <- c("ClusterID", "REV_Counts")

rev_100_95_v_musc_URDB_info <- read.table("49615_rev_mapped_mcy_100_95.muscato_info.txt", sep='\t', header=FALSE)
colnames(rev_100_95_v_musc_URDB_info) <- c("ClusterID", "Gene_name", "Gene_info", "Source_LCA", "Worst_case_LCA", 
                                      "count match at 95%", "LCA_100_pid", "LCA_95_pid", "contamination_flag", "KEGG", "PFAM",
                                      "COG", "GO", "Interpro", "EC", "Metal_binding", "location", "reaction", "metacyc_id",
                                      "rxn_direction", "reactants", "products", "protein_seq", "gene_seq")

#Filter out unneeded columns in the info data frames:
#These are the columns that I want
keep <- c("ClusterID", "Gene_name", "LCA_100_pid", "LCA_95_pid", "products", "gene_seq")
#Alter the data tables so that they only contain those columns:  
fwd_100_95_v_musc_URDB_info <- fwd_100_95_v_musc_URDB_info[ , colnames(fwd_100_95_v_musc_URDB_info) %in% keep ]
rev_100_95_v_musc_URDB_info <- rev_100_95_v_musc_URDB_info[ , colnames(rev_100_95_v_musc_URDB_info) %in% keep ]

#Remove the reverse "_r" hits:
#Get all of the entries that don't have the _r flag
keep <- grep("_r", fwd_100_95_v_musc_URDB_counts$ClusterID, invert = TRUE, value = TRUE)
#Only save rows in keep
fwd_100_95_v_musc_URDB_counts <- fwd_100_95_v_musc_URDB_counts[ fwd_100_95_v_musc_URDB_counts$ClusterID %in% keep , ]
#Repeat below for the reverse reads
keep <- grep("_r", rev_100_95_v_musc_URDB_counts$ClusterID, invert = TRUE, value = TRUE)
rev_100_95_v_musc_URDB_counts <- rev_100_95_v_musc_URDB_counts[ rev_100_95_v_musc_URDB_counts$ClusterID %in% keep, ]

#For some reason both count files have an empty 3rd column, so I'm going to remove it from each dataframe:
fwd_100_95_v_musc_URDB_counts <- fwd_100_95_v_musc_URDB_counts[ , 1:2]
rev_100_95_v_musc_URDB_counts <- rev_100_95_v_musc_URDB_counts[ , 1:2]

#Merge the count and info dataframes:
fwd_100_95_v_musc_URDB_merged <- merge(fwd_100_95_v_musc_URDB_counts, fwd_100_95_v_musc_URDB_info, by="ClusterID", all=TRUE)
rev_100_95_v_musc_URDB_merged <- merge(rev_100_95_v_musc_URDB_counts, rev_100_95_v_musc_URDB_info, by="ClusterID", all=TRUE)

#Merge forward and reverse dataframes:
total_100_95_v_musc_URDB_merged <- merge(fwd_100_95_v_musc_URDB_merged, rev_100_95_v_musc_URDB_merged, by=c("ClusterID", "Gene_name", "LCA_100_pid", "LCA_95_pid", "products", "gene_seq"), all=TRUE)

#Set NAs in the read count columns in the merged dataframes to 0:
total_100_95_v_musc_URDB_merged$FWD_Counts[is.na(total_100_95_v_musc_URDB_merged$FWD_Counts)] <- 0
total_100_95_v_musc_URDB_merged$REV_Counts[is.na(total_100_95_v_musc_URDB_merged$REV_Counts)] <- 0

#Sum the FWD and REV counts:
total_100_95_v_musc_URDB_merged$Total_Counts <- total_100_95_v_musc_URDB_merged$FWD_Counts + total_100_95_v_musc_URDB_merged$REV_Counts

#Export the final results to a tab-delimited table file:  
write.table(total_100_95_v_musc_URDB_merged, file="total_100_95_v_musc_URDB.txt", quote = FALSE, sep="\t")
```

Manually fix how the columns are labeled and removed the column that holds row numbers.  
Generate a fasta file from the table with the gene sequences:  
```{bash}
#Remove any previously generated fasta files with the same name (we want to overwrite with this script, not append to the old file): 
#true command tells the script not to stop if the command fails.
rm total_100_95_v_musc_URDB.fasta | true

#Extract the URDB database sequence IDs:  
awk -F '\t' 'NR>1 {print $1}' total_100_95_v_musc_URDB.txt > fasta_headers.list

#For each fasta header, print the header, followed by the corresponding sequence:
for i in `cat fasta_headers.list`; do
  echo $i | sed 's/:/_/g' | sed 's/^/>/g' >> total_100_95_v_musc_URDB.fasta
  awk -F '\t' '{if($1 == "'$i'") {print $6}}' total_100_95_v_musc_URDB.txt >> total_100_95_v_musc_URDB.fasta
done
```
I blast the gene sequences in the URDB where reads mapped with MUSCATO to gene sequences in closed Microcystis genomes.  Many of the generic annotations in the URDB are actually mcy genes (or fragments of mcy genes).  

Note that the reads that mapped to the mcy at 95-90% identity did not have significant hits to URDB with muscato (which is very stringent):  

What proportion of the reads that mapped to the mcy cluster at 100-95% ID are mapping to other genes in URDB with muscato?  
```{r}
#Sum the data by gene_name:
Muscato_100_95_summed_by_gene_name <- total_100_95_v_musc_URDB_merged %>%
  group_by(Gene_name) %>%
  summarise(Total=sum(Total_Counts))

#Create a bar graph of the Total number of reads mapped to the different gene categories:
top25_100_95_musccato_gene_plot <- Muscato_100_95_summed_by_gene_name %>%
                            arrange(desc(Total)) %>%
                              slice(1:25) %>%
                                ggplot(aes(x=reorder(Gene_name, -Total), y=Total)) +
                                  geom_bar(stat = "identity", position = position_dodge()) +
                                  theme_classic() +
                                  ggtitle("Top 25 functions") +
                                  theme(
                                    axis.line.x = element_line(size=0.1),
                                    axis.line.y = element_line(size=0.1),
                                    axis.title.x = element_text(size=12, margin = margin(t = 10, r = 0, b = 0, l = 0)), 
                                    axis.title.y = element_text(size=12, margin = margin(t = 0, r = 10, b = 0, l = 0)),
                                    panel.grid.major = element_blank(),
                                    panel.grid.minor = element_blank(),
                                    axis.text.x = element_text(size = 6, color = "black", angle = 45, vjust = 1, hjust = 1,
                                                               margin = margin(t = 10, r = 0, b = 0, l = 0)),
                                    axis.text.y = element_text(size = 10, color = "black", margin = margin(t = 0, r = 10, b = 0, l = 0)),
                                    axis.ticks.length = unit(-0.1, "cm"),
                                    axis.ticks = element_line(size = 0.1)
                                    ) +
                                  ylab("Total reads mapped") +
                                  xlab("") 


Second_chunk_100_95_muscato_gene_plot <- Muscato_100_95_summed_by_gene_name %>%
                                    arrange(desc(Total)) %>%
                                      slice(26:50) %>%
                                        ggplot(aes(x=reorder(Gene_name, -Total), y=Total)) +
                                          geom_bar(stat = "identity", position = position_dodge()) +
                                          theme_classic() +
                                          ggtitle("Next 25 functions") +
                                          theme(
                                            axis.line.x = element_line(size=0.1),
                                            axis.line.y = element_line(size=0.1),
                                            axis.title.x = element_text(size=12, margin = margin(t = 10, r = 0, b = 0, l = 0)), 
                                            axis.title.y = element_text(size=12, margin = margin(t = 0, r = 10, b = 0, l = 0)),
                                            panel.grid.major = element_blank(),
                                            panel.grid.minor = element_blank(),
                                            axis.text.x = element_text(size = 6, color = "black", angle = 45, vjust = 1, hjust = 1,
                                                                       margin = margin(t = 10, r = 0, b = 0, l = 0)),
                                            axis.text.y = element_text(size = 10, color = "black", margin = margin(t = 0, r = 10, b = 0, l = 0)),
                                            axis.ticks.length = unit(-0.1, "cm"),
                                            axis.ticks = element_line(size = 0.1)
                                            ) +
                                          ylab("Total reads mapped") +
                                          xlab("") 

total_100_95_muscato_gene_plot <- top25_100_95_musccato_gene_plot + Second_chunk_100_95_muscato_gene_plot
total_100_95_muscato_gene_plot
ggsave("total_100_95_muscato_gene_plot.pdf", plot = total_100_95_muscato_gene_plot, width = 22, height = 18, units = "cm", dpi=320)
```
What proportion of the reads that aligned to the Microcystis mcy cluster at 100-95 % identity are aligning to other organisms in URDB with muscato?
```{r}
#Sum the data by LCA:
total_100_95_muscato_summed_by_LCA <- total_100_95_v_musc_URDB_merged %>%
  group_by(LCA_100_pid) %>%
  summarise(Total=sum(Total_Counts))

#Divide the LCA taxonomy into different taxonomic ranks:
#Note that there is an unequal number of ranks listed for each row because Teal's pipeline only lists up to the lowest possible taxonomic rank So this command will generate errors about not enough fields expected. This is OK, the fields with missing data will simply be empty.
total_100_95_muscato_summed_by_LCA <- separate(total_100_95_muscato_summed_by_LCA, LCA_100_pid, c("Domain", "Phylum", "Class", "Order", "Family", "Genus", "Species"), sep=";", remove=TRUE)

#Transpose the table
total_100_95_muscato_summed_by_LCA <- t(total_100_95_muscato_summed_by_LCA)

#Fill the NAs in the taxonomy with the last non-NA value, the reorient to the correct position:
total_100_95_muscato_summed_by_LCA <- na.locf(total_100_95_muscato_summed_by_LCA)
total_100_95_muscato_summed_by_LCA <- as.data.frame(t(total_100_95_muscato_summed_by_LCA))
total_100_95_muscato_summed_by_LCA$Total <- as.numeric(total_100_95_muscato_summed_by_LCA$Total)

#Create a bar graph of the Total number of reads mapped to the different gene categories:
Muscato_P100_95_LCA_plot <- total_100_95_muscato_summed_by_LCA %>%
                            arrange(desc(Total)) %>%
                                ggplot(aes(x=reorder(Genus, -Total), y=Total)) +
                                  geom_bar(stat = "identity", position = position_dodge()) +
                                  theme_classic() +
                                  theme(
                                    axis.line.x = element_line(size=0.1),
                                    axis.line.y = element_line(size=0.1),
                                    axis.title.x = element_text(size=12, margin = margin(t = 10, r = 0, b = 0, l = 0)), 
                                    axis.title.y = element_text(size=12, margin = margin(t = 0, r = 10, b = 0, l = 0)),
                                    panel.grid.major = element_blank(),
                                    panel.grid.minor = element_blank(),
                                    axis.text.x = element_text(size = 6, color = "black", angle = 45, vjust = 1, hjust = 1,
                                                               margin = margin(t = 10, r = 0, b = 0, l = 0)),
                                    axis.text.y = element_text(size = 10, color = "black", margin = margin(t = 0, r = 10, b = 0, l = 0)),
                                    axis.ticks.length = unit(-0.1, "cm"),
                                    axis.ticks = element_line(size = 0.1)
                                    ) +
                                  ylab("Total reads mapped") +
                                  xlab("Taxon") 

Muscato_P100_95_LCA_plot
ggsave("Muscato_P100_95_LCA_plot.pdf", plot = Muscato_P100_95_LCA_plot, width = 22, height = 18, units = "cm", dpi=320)
```
It looks like the answer when aligning the reads to the URDB with Muscato is more or less the same as when aligning them with Diamond. There is mapping to mcy and various other genes within Microcystis. Mapping to other organisms is not really happening at any significant numbers.

Make a histogram of read start locations on the mcy operon for strain 7806:  
```{r}
#Load the dataframe
PCC_7806_only_BLAST <- read.table("Sample_49615_cat_PCC7806_mcy.blastn.txt", sep='\t', header = FALSE)
colnames(PCC_7806_only_BLAST) <- c("Query", "Subject", "PID", "Aln_Len", "Mismatches", "Gaps", "QStart", "QStop", "SStart", "SStop",
                                   "Eval", "BitScore", "Qlen", "Slen", "Qcovs")

#Only keep hits to mcyD:
PCC_7806_mcyD_only_BLAST <- PCC_7806_only_BLAST[ PCC_7806_only_BLAST$Subject == "McyD_Microcystis_aeruginosa_PCC_7806", ]

#Histogram of Subject start and stop:  
PCC7806_mcyD_align_coord_histo <- ggplot(PCC_7806_mcyD_only_BLAST, aes(x=SStart)) +
                              geom_histogram(breaks=seq(1,11734, by = 4), color="black", fill="white") +
                              scale_x_continuous(breaks = seq(1,11734, by = 500)) +
                              theme_classic() +
                              theme(
                                axis.line.x = element_line(size=0.1),
                                axis.line.y = element_line(size=0.1),
                                axis.title.x = element_text(size=12, margin = margin(t = 10, r = 0, b = 0, l = 0)), 
                                axis.title.y = element_text(size=12, margin = margin(t = 0, r = 10, b = 0, l = 0)),
                                panel.grid.major = element_blank(),
                                panel.grid.minor = element_blank(),
                                axis.text.x = element_text(size = 12, color = "black", angle = 45, vjust = 1, hjust = 1,
                                                           margin = margin(t = 10, r = 0, b = 0, l = 0)),
                                axis.text.y = element_text(size = 12, color = "black", margin = margin(t = 0, r = 10, b = 0, l = 0)),
                                axis.ticks.length = unit(-0.1, "cm"),
                                axis.ticks = element_line(size = 0.1)
                                ) +
                              ylab("Count") +
                              xlab("mcyD coordinate")

#Used this piece of code to zoom in on areas were the coverage spiked:  
zoom_PCC7806_mcyD_align_coord_histo <- ggplot(PCC_7806_mcyD_only_BLAST, aes(x=SStart)) +
                              geom_histogram(breaks=seq(8000,8500, by = 4), color="black", fill="white") +
                              scale_x_continuous(breaks = seq(8000,8500, by = 8)) +
                              theme_classic() +
                              theme(
                                axis.line.x = element_line(size=0.1),
                                axis.line.y = element_line(size=0.1),
                                axis.title.x = element_text(size=12, margin = margin(t = 10, r = 0, b = 0, l = 0)), 
                                axis.title.y = element_text(size=12, margin = margin(t = 0, r = 10, b = 0, l = 0)),
                                panel.grid.major = element_blank(),
                                panel.grid.minor = element_blank(),
                                axis.text.x = element_text(size = 6, color = "black", angle = 45, vjust = 1, hjust = 1,
                                                           margin = margin(t = 10, r = 0, b = 0, l = 0)),
                                axis.text.y = element_text(size = 12, color = "black", margin = margin(t = 0, r = 10, b = 0, l = 0)),
                                axis.ticks.length = unit(-0.1, "cm"),
                                axis.ticks = element_line(size = 0.1)
                                ) +
                              ylab("Count") +
                              xlab("mcyD coordinate")

PCC7806_mcyD_align_coord_histo
zoom_PCC7806_mcyD_align_coord_histo
```
Make a histogram of mapped read lengths, because I am consider that some of the higher mcy abudnance than expected may be from short reads that share high percent identity are actually from other organisms that have very small regions of high sequence similarity with Microcystis mcy genes:    
```{r}
PCC7806_mcy_read_lengths <- ggplot(PCC_7806_only_BLAST, aes(x=Qlen)) +
                              geom_histogram(breaks=seq(75,125, by = 5), color="black", fill="white") +
                              scale_x_continuous(breaks = seq(75,125, by = 5)) +
                              ggtitle("PCC7806 mcy cluster mapped >= 95% ID, >= 80% read coverage") +
                              theme_classic() +
                              theme(
                                axis.line.x = element_line(size=0.1),
                                axis.line.y = element_line(size=0.1),
                                axis.title.x = element_text(size=12, margin = margin(t = 10, r = 0, b = 0, l = 0)), 
                                axis.title.y = element_text(size=12, margin = margin(t = 0, r = 10, b = 0, l = 0)),
                                panel.grid.major = element_blank(),
                                panel.grid.minor = element_blank(),
                                axis.text.x = element_text(size = 12, color = "black", angle = 45, vjust = 1, hjust = 1,
                                                           margin = margin(t = 10, r = 0, b = 0, l = 0)),
                                axis.text.y = element_text(size = 12, color = "black", margin = margin(t = 0, r = 10, b = 0, l = 0)),
                                axis.ticks.length = unit(-0.1, "cm"),
                                axis.ticks = element_line(size = 0.1)
                                ) +
                              ylab("Count") +
                              xlab("Read Length (bp)")

PCC7806_mcy_read_lengths
ggsave("PCC7806_mcy_read_lengths.pdf", plot = PCC7806_mcy_read_lengths, width = 22, height = 18, units = "cm", dpi=320)
```
Most of the reads are above 100 bp, so this is likely not a concern.  

There was some concern about whether the results of aligning the mcy-mapped reads to Teal's URDB is really comparable because I had to use Muscato (stringent aligner) and diamond blastx (DNA-protein rather than DNA-DNA). So I also did a blastn to align the reads to NCBI's nt database.  



16S read mapping
---

Now let's look at what might be causing differences between the 16S reads that mapped with BBmap vs those that mapped with BLAST:

Approach: I extracted the reads that mapped to the Microcystis V4 reference with bbmap (which was ran by Colleen Yancey), but not with BLAST (which was run by Derek Smith). The reference database and alignment cutoffs were supposedly the same. Although there may not have been an alignment length cutoff with bbmap (we are still figuring that out).  

I took those reads and aligned them to Silva small subunit database v 138, allowing for multi-mapping (i.e. I counted all the hits that a single read had, so a single read may be tallied multiple times). I also aligned all the reads that mapped with BLAST to Silva, just to ensure that Microcystis was indeed the best match for those reads.  

I'm only looking at one sample, which had the largest discrepency in % toxic Microcystis cells using either BLAST or BBmap.  

What proportion of the reads were mapping to Microcystis vs. other organisms?  
```{r}
#First load in the dataframe of counts to Silva for bbmapped and BLAST-mapped reads:
BBMapped_only_v_Silva <- read.table("Read_counts_42895_bbmap_only_Microcystis_V4_vs_Silva_v138.blastn.postblast.txt", sep='\t', header=FALSE)
BLAST_mapped_v_Silva <- read.table("Read_counts_42895_BLAST_mapped_Microcystis_V4_vs_Silva_v138.blastn.postblast.txt", sep='\t', header=FALSE)

#Label the columns:
colnames(BBMapped_only_v_Silva) <- c("Taxonomy", "Counts")
colnames(BLAST_mapped_v_Silva) <- c("Taxonomy", "Counts")

#Sum the data by Taxonomy:
BBMapped_only_v_Silva_summary <- BBMapped_only_v_Silva %>%
  group_by(Taxonomy) %>%
  summarise(Total=sum(Counts))

BLAST_mapped_v_Silva_summary <- BLAST_mapped_v_Silva %>%
  group_by(Taxonomy) %>%
  summarise(Total=sum(Counts))

#Divide the taxonomy into different taxonomic ranks:
#Note that there is an unequal number of ranks listed for each row because Teal's pipeline only lists up to the lowest possible taxonomic rank So this command will generate errors about not enough fields expected. This is OK, the fields with missing data will simply be empty.
BBMapped_only_v_Silva_summary <- separate(BBMapped_only_v_Silva_summary, Taxonomy, c("Domain", "Phylum", "Class", "Order", "Family", "Genus"), sep=";", remove=TRUE)

BLAST_mapped_v_Silva_summary <- separate(BLAST_mapped_v_Silva_summary, Taxonomy, c("Domain", "Phylum", "Class", "Order", "Family", "Genus"), sep=";", remove=TRUE)

#Transpose the tables
BBMapped_only_v_Silva_summary <- t(BBMapped_only_v_Silva_summary)
BLAST_mapped_v_Silva_summary <- t(BLAST_mapped_v_Silva_summary)

#Fill the NAs in the taxonomy with the last non-NA value, then reorient to the correct position:
BBMapped_only_v_Silva_summary <- na.locf(BBMapped_only_v_Silva_summary)
BBMapped_only_v_Silva_summary <- as.data.frame(t(BBMapped_only_v_Silva_summary))
BBMapped_only_v_Silva_summary$Total <- as.numeric(BBMapped_only_v_Silva_summary$Total)

BLAST_mapped_v_Silva_summary <- na.locf(BLAST_mapped_v_Silva_summary)
BLAST_mapped_v_Silva_summary <- as.data.frame(t(BLAST_mapped_v_Silva_summary))
BLAST_mapped_v_Silva_summary$Total <- as.numeric(BLAST_mapped_v_Silva_summary$Total)
```

How many of the BBmapped reads mapped to other organisms in Silva?  
```{r}
BBmap_16S_V4_LCA_plot <- BBMapped_only_v_Silva_summary %>%
                            arrange(desc(Total)) %>%
                                ggplot(aes(x=reorder(Genus, -Total), y=Total)) +
                                  geom_bar(stat = "identity", position = position_dodge()) +
                                  theme_classic() +
                                  theme(
                                    axis.line.x = element_line(size=0.1),
                                    axis.line.y = element_line(size=0.1),
                                    axis.title.x = element_text(size=12, margin = margin(t = 10, r = 0, b = 0, l = 0)), 
                                    axis.title.y = element_text(size=12, margin = margin(t = 0, r = 10, b = 0, l = 0)),
                                    panel.grid.major = element_blank(),
                                    panel.grid.minor = element_blank(),
                                    axis.text.x = element_text(size = 6, color = "black", angle = 45, vjust = 1, hjust = 1,
                                                               margin = margin(t = 10, r = 0, b = 0, l = 0)),
                                    axis.text.y = element_text(size = 10, color = "black", margin = margin(t = 0, r = 10, b = 0, l = 0)),
                                    axis.ticks.length = unit(-0.1, "cm"),
                                    axis.ticks = element_line(size = 0.1)
                                    ) +
                                  ylab("Total reads mapped") +
                                  xlab("Taxon") 

BBmap_16S_V4_LCA_plot
ggsave("BBmap_16S_V4_LCA_plot.pdf", plot = BBmap_16S_V4_LCA_plot, width = 22, height = 18, units = "cm", dpi=320)
```
How many of the BBmapped reads mapped to other organisms in Silva?
```{r}
BLAST_16S_V4_LCA_plot <- BLAST_mapped_v_Silva_summary %>%
                            arrange(desc(Total)) %>%
                                ggplot(aes(x=reorder(Genus, -Total), y=Total)) +
                                  geom_bar(stat = "identity", position = position_dodge()) +
                                  theme_classic() +
                                  theme(
                                    axis.line.x = element_line(size=0.1),
                                    axis.line.y = element_line(size=0.1),
                                    axis.title.x = element_text(size=12, margin = margin(t = 10, r = 0, b = 0, l = 0)), 
                                    axis.title.y = element_text(size=12, margin = margin(t = 0, r = 10, b = 0, l = 0)),
                                    panel.grid.major = element_blank(),
                                    panel.grid.minor = element_blank(),
                                    axis.text.x = element_text(size = 6, color = "black", angle = 45, vjust = 1, hjust = 1,
                                                               margin = margin(t = 10, r = 0, b = 0, l = 0)),
                                    axis.text.y = element_text(size = 10, color = "black", margin = margin(t = 0, r = 10, b = 0, l = 0)),
                                    axis.ticks.length = unit(-0.1, "cm"),
                                    axis.ticks = element_line(size = 0.1)
                                    ) +
                                  ylab("Total reads mapped") +
                                  xlab("Taxon") 

BLAST_16S_V4_LCA_plot
ggsave("BLAST_16S_V4_LCA_plot.pdf", plot = BLAST_16S_V4_LCA_plot, width = 22, height = 18, units = "cm", dpi=320)
```
Most of the reads that mapped in BBMap had better hits to Chloroplasts when aligned to Silva. I wonder if there was no alignment length cutoff with BBmap when the reads were tallied?

Conversely, most of the reads that mapped using BLAST had best hits to Microcystis in Silva. There was very little non-specific mapping to other organisms with both methods.

Next, I want to check to see if any reads mapped to the Microcystis V4 reference at <97% ID without having higher best matches to other organisms.  

What is the percent identity distribution of read best matches to the Microcystis V4 reference?  
```{r}
FWD_V4_BLAST_no_pid_cutoff <- read.table("Sample_42895_adtrim_clean_qtrim_fwd_vs_16S.blastn.no_id_cutoff.postblast", sep='\t', header=FALSE)
colnames(FWD_V4_BLAST_no_pid_cutoff) <- c("Read", "Reference", "PID", "Aln_Len", "Mismatches", "Gaps", "QStart", "QStop", "RefStart",
                                          "RefStop", "Eval", "BitScore", "Read_Len", "Ref_Len", "Read_Cov")


REV_V4_BLAST_no_pid_cutoff <- read.table("Sample_42895_adtrim_clean_qtrim_rev_vs_16S.blastn.no_id_cutoff.postblast", sep='\t', header=FALSE)
colnames(REV_V4_BLAST_no_pid_cutoff) <- c("Read", "Reference", "PID", "Aln_Len", "Mismatches", "Gaps", "QStart", "QStop", "RefStart",
                                          "RefStop", "Eval", "BitScore", "Read_Len", "Ref_Len", "Read_Cov")

#Remove alignments not to Microcystis or with coverage below 80%:  
FWD_V4_BLAST_no_pid_cutoff <- filter(FWD_V4_BLAST_no_pid_cutoff, Reference == "Microcystis_Berry_2017_Otu00005" & Read_Cov >= 80)
REV_V4_BLAST_no_pid_cutoff <- filter(REV_V4_BLAST_no_pid_cutoff, Reference == "Microcystis_Berry_2017_Otu00005" & Read_Cov >= 80)

#Make a histogram of %ID in the forward and reverse hits:  
FWD_Microcystis_hit_histo <- ggplot(FWD_V4_BLAST_no_pid_cutoff, aes(x=PID)) +
                              geom_histogram(breaks=seq(90,100, by = 1), color="black", fill="white") +
                              scale_x_continuous(breaks = seq(90,100, by = 1)) +
                              ggtitle("Forward reads") +
                              theme_classic() +
                              theme(
                                axis.line.x = element_line(size=0.1),
                                axis.line.y = element_line(size=0.1),
                                axis.title.x = element_text(size=12, margin = margin(t = 10, r = 0, b = 0, l = 0)), 
                                axis.title.y = element_text(size=12, margin = margin(t = 0, r = 10, b = 0, l = 0)),
                                panel.grid.major = element_blank(),
                                panel.grid.minor = element_blank(),
                                axis.text.x = element_text(size = 10, color = "black", angle = 45, vjust = 1, hjust = 1,
                                                           margin = margin(t = 10, r = 0, b = 0, l = 0)),
                                axis.text.y = element_text(size = 12, color = "black", margin = margin(t = 0, r = 10, b = 0, l = 0)),
                                axis.ticks.length = unit(-0.1, "cm"),
                                axis.ticks = element_line(size = 0.1)
                                ) +
                              ylab("Count") +
                              xlab("Percent Identity to Microcystis Reference")

REV_Microcystis_hit_histo <- ggplot(REV_V4_BLAST_no_pid_cutoff, aes(x=PID)) +
                              geom_histogram(breaks=seq(90,100, by = 1), color="black", fill="white") +
                              scale_x_continuous(breaks = seq(90,100, by = 1)) +
                              ggtitle("Reverse reads") +
                              theme_classic() +
                              theme(
                                axis.line.x = element_line(size=0.1),
                                axis.line.y = element_line(size=0.1),
                                axis.title.x = element_text(size=12, margin = margin(t = 10, r = 0, b = 0, l = 0)), 
                                axis.title.y = element_text(size=12, margin = margin(t = 0, r = 10, b = 0, l = 0)),
                                panel.grid.major = element_blank(),
                                panel.grid.minor = element_blank(),
                                axis.text.x = element_text(size = 10, color = "black", angle = 45, vjust = 1, hjust = 1,
                                                           margin = margin(t = 10, r = 0, b = 0, l = 0)),
                                axis.text.y = element_text(size = 12, color = "black", margin = margin(t = 0, r = 10, b = 0, l = 0)),
                                axis.ticks.length = unit(-0.1, "cm"),
                                axis.ticks = element_line(size = 0.1)
                                ) +
                              ylab("Count") +
                              xlab("Percent Identity to Microcystis Reference")

All_Microcystis_hit_histo_42895 <- FWD_Microcystis_hit_histo + REV_Microcystis_hit_histo
All_Microcystis_hit_histo_42895
ggsave("All_Microcystis_hit_histo_42895.pdf", plot = All_Microcystis_hit_histo_42895, width = 22, height = 18, units = "cm", dpi=320)
```
There is one reverse read that matched the reference at 94.85% identity. All of the best matches to Silva were also at this cutoff threshold, with no ambiguous hits to other organisms.  

Next I want to check for true Microcystis hits at lower percent identity in a few samples with higher Microcystis abundance (are we really missing a lot of reads?).  
```{r}
FWD49615_V4_BLAST_no_pid_cutoff <- read.table("Sample_49615_adtrim_clean_qtrim_fwd_vs_16S.blastn.no_id_cutoff.postblast", sep='\t', header=FALSE)
colnames(FWD49615_V4_BLAST_no_pid_cutoff ) <- c("Read", "Reference", "PID", "Aln_Len", "Mismatches", "Gaps", "QStart", "QStop", "RefStart",
                                          "RefStop", "Eval", "BitScore", "Read_Len", "Ref_Len", "Read_Cov")


REV49615_V4_BLAST_no_pid_cutoff <- read.table("Sample_49615_adtrim_clean_qtrim_rev_vs_16S.blastn.no_id_cutoff.postblast", sep='\t', header=FALSE)
colnames(REV49615_V4_BLAST_no_pid_cutoff) <- c("Read", "Reference", "PID", "Aln_Len", "Mismatches", "Gaps", "QStart", "QStop", "RefStart",
                                          "RefStop", "Eval", "BitScore", "Read_Len", "Ref_Len", "Read_Cov")

#Remove alignments not to Microcystis or with coverage below 80%:  
FWD49615_V4_BLAST_no_pid_cutoff <- filter(FWD49615_V4_BLAST_no_pid_cutoff, Reference == "Microcystis_Berry_2017_Otu00005" & Read_Cov >= 80)
REV49615_V4_BLAST_no_pid_cutoff <- filter(REV49615_V4_BLAST_no_pid_cutoff, Reference == "Microcystis_Berry_2017_Otu00005" & Read_Cov >= 80)

#Make a histogram of %ID in the forward and reverse hits:  
FWD49615_Microcystis_hit_histo <- ggplot(FWD49615_V4_BLAST_no_pid_cutoff, aes(x=PID)) +
                              geom_histogram(breaks=seq(90,100, by = 1), color="black", fill="white") +
                              scale_x_continuous(breaks = seq(90,100, by = 1)) +
                              ggtitle("Forward reads") +
                              theme_classic() +
                              theme(
                                axis.line.x = element_line(size=0.1),
                                axis.line.y = element_line(size=0.1),
                                axis.title.x = element_text(size=12, margin = margin(t = 10, r = 0, b = 0, l = 0)), 
                                axis.title.y = element_text(size=12, margin = margin(t = 0, r = 10, b = 0, l = 0)),
                                panel.grid.major = element_blank(),
                                panel.grid.minor = element_blank(),
                                axis.text.x = element_text(size = 10, color = "black", angle = 45, vjust = 1, hjust = 1,
                                                           margin = margin(t = 10, r = 0, b = 0, l = 0)),
                                axis.text.y = element_text(size = 12, color = "black", margin = margin(t = 0, r = 10, b = 0, l = 0)),
                                axis.ticks.length = unit(-0.1, "cm"),
                                axis.ticks = element_line(size = 0.1)
                                ) +
                              ylab("Count") +
                              xlab("Percent Identity to Microcystis Reference")

REV49615_Microcystis_hit_histo <- ggplot(REV49615_V4_BLAST_no_pid_cutoff, aes(x=PID)) +
                              geom_histogram(breaks=seq(90,100, by = 1), color="black", fill="white") +
                              scale_x_continuous(breaks = seq(90,100, by = 1)) +
                              ggtitle("Reverse reads") +
                              theme_classic() +
                              theme(
                                axis.line.x = element_line(size=0.1),
                                axis.line.y = element_line(size=0.1),
                                axis.title.x = element_text(size=12, margin = margin(t = 10, r = 0, b = 0, l = 0)), 
                                axis.title.y = element_text(size=12, margin = margin(t = 0, r = 10, b = 0, l = 0)),
                                panel.grid.major = element_blank(),
                                panel.grid.minor = element_blank(),
                                axis.text.x = element_text(size = 10, color = "black", angle = 45, vjust = 1, hjust = 1,
                                                           margin = margin(t = 10, r = 0, b = 0, l = 0)),
                                axis.text.y = element_text(size = 12, color = "black", margin = margin(t = 0, r = 10, b = 0, l = 0)),
                                axis.ticks.length = unit(-0.1, "cm"),
                                axis.ticks = element_line(size = 0.1)
                                ) +
                              ylab("Count") +
                              xlab("Percent Identity to Microcystis Reference")

All_Microcystis_hit_histo_49615 <- FWD49615_Microcystis_hit_histo + REV49615_Microcystis_hit_histo
All_Microcystis_hit_histo_49615
ggsave("All_Microcystis_hit_histo_49615.pdf", plot = All_Microcystis_hit_histo_49615, width = 22, height = 18, units = "cm", dpi=320)
```
There are best hits to the Microcystis V4 reference below 97 % ID in this sample too, but are they all best matches to Microcystis?  
I extracted the reads that mapped to the Microcystis V4 (as best match), and aligned them to Silva SSU database:  
Then, make a histogram of the percent identity of the matches of those reads to Microcystis:  
```{r}
#Import the matches to Silva:
FWD_Mapped_49615_vs_Silva <- read.table("49615_fwd_mapped_Microcystis_V4_vs_Silva_v138.blastn.postblast", sep="\t", header=TRUE)
REV_Mapped_49615_vs_Silva <- read.table("49615_rev_mapped_Microcystis_V4_vs_Silva_v138.blastn.postblast", sep="\t", header=TRUE)

#Tidy up the taxonomy columns:  
FWD_Mapped_49615_vs_Silva <- separate(FWD_Mapped_49615_vs_Silva, Subject, c("Subject", "Domain", "Phylum", "Class", "Order", "Family", "Genus"), sep=";", remove=TRUE)
REV_Mapped_49615_vs_Silva <- separate(REV_Mapped_49615_vs_Silva, Subject, c("Subject", "Domain", "Phylum", "Class", "Order", "Family", "Genus"), sep=";", remove=TRUE)

#Make NAs in taxonomy columns be listed as unknown: 
FWD_Mapped_49615_vs_Silva[is.na(FWD_Mapped_49615_vs_Silva)] <- "unknown"
REV_Mapped_49615_vs_Silva[is.na(REV_Mapped_49615_vs_Silva)] <- "unknown"

#If the reference genus is not Microcystis, list it as "non_Microcystis" for plotting purposes:
FWD_Mapped_49615_vs_Silva$Genus[FWD_Mapped_49615_vs_Silva$Genus != "Microcystis_PCC7914"] <- "non_Microcystis"
REV_Mapped_49615_vs_Silva$Genus[REV_Mapped_49615_vs_Silva$Genus != "Microcystis_PCC7914"] <- "non_Microcystis"

#Remove all of the reads that matched to Silva with Read Coverage below 80% of read length:
FWD_Mapped_49615_vs_Silva <- FWD_Mapped_49615_vs_Silva[ FWD_Mapped_49615_vs_Silva$Qcovs >= 80 , ]
REV_Mapped_49615_vs_Silva <- REV_Mapped_49615_vs_Silva[ REV_Mapped_49615_vs_Silva$Qcovs >= 80, ]

#Get a list of reads that mapped to the Microcystis V4 at >= 97% ID:  
FWD49615_readlitst_100_97 <- FWD49615_V4_BLAST_no_pid_cutoff[ FWD49615_V4_BLAST_no_pid_cutoff$PID >= 97 , 1]
REV49615_readlitst_100_97 <- REV49615_V4_BLAST_no_pid_cutoff[ REV49615_V4_BLAST_no_pid_cutoff$PID >= 97 , 1]

#Give a histogram of the % matches to Silva (counting all hits, each read counted multiple times):    
FWD_Mapped_49615_100_97 <- filter(FWD_Mapped_49615_vs_Silva, Query %in% FWD49615_readlitst_100_97) %>%
  ggplot(aes(x=Percent_ID, fill=Genus)) +
    geom_histogram(breaks=seq(75,100, by = 1), color = "#e9ecef", alpha=0.6, position = 'identity') +
    scale_x_continuous(breaks = seq(75,100, by = 1)) +
    scale_fill_manual(values=c("#69b3a2", "#404080")) +
    ggtitle("Forward reads") +
    theme_classic() +
    theme(
      axis.line.x = element_line(size=0.1),
      axis.line.y = element_line(size=0.1),
      axis.title.x = element_text(size=12, margin = margin(t = 10, r = 0, b = 0, l = 0)), 
      axis.title.y = element_text(size=12, margin = margin(t = 0, r = 10, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 10, color = "black", angle = 45, vjust = 1, hjust = 1,
                                 margin = margin(t = 10, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 12, color = "black", margin = margin(t = 0, r = 10, b = 0, l = 0)),
      axis.ticks.length = unit(-0.1, "cm"),
      axis.ticks = element_line(size = 0.1),
      legend.position = "none",
      ) +
    ylab("Count") +
    xlab("Percent Match to Reference")

REV_Mapped_49615_100_97 <- filter(REV_Mapped_49615_vs_Silva, Query %in% REV49615_readlitst_100_97) %>%
  ggplot(aes(x=Percent_ID, fill=Genus)) +
    geom_histogram(breaks=seq(75,100, by = 1),  color = "#e9ecef", alpha=0.6, position = 'identity') +
    scale_x_continuous(breaks = seq(75,100, by = 1)) +
    scale_fill_manual(values=c("#69b3a2", "#404080")) +
    ggtitle("Reverse reads") +
    theme_classic() +
    theme(
      axis.line.x = element_line(size=0.1),
      axis.line.y = element_line(size=0.1),
      axis.title.x = element_text(size=12, margin = margin(t = 10, r = 0, b = 0, l = 0)), 
      axis.title.y = element_text(size=12, margin = margin(t = 0, r = 10, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 10, color = "black", angle = 45, vjust = 1, hjust = 1,
                                 margin = margin(t = 10, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 12, color = "black", margin = margin(t = 0, r = 10, b = 0, l = 0)),
      axis.ticks.length = unit(-0.1, "cm"),
      axis.ticks = element_line(size = 0.1)
      ) +
    ylab("Count") +
    xlab("Percent Match to Reference")

All_mapped_49615_100_97 <- FWD_Mapped_49615_100_97 + REV_Mapped_49615_100_97 + guide_area()
All_mapped_49615_100_97
ggsave("All_mapped_49615_100_97.pdf", plot = All_mapped_49615_100_97, width = 40, height = 18, units = "cm", dpi=320)
```
The error is generated because some taxonomies are not filled out all the way (for example, uncultured or unclassified sequences in Silva).  

Make another histogram of percent match to Silva, but for reads that mapped to Microcystis at 97-94 percent identity:  
```{r}
#Get a list of reads that mapped to the Microcystis V4 at 97-94 % ID:  
FWD49615_readlitst_97_94 <- FWD49615_V4_BLAST_no_pid_cutoff[ FWD49615_V4_BLAST_no_pid_cutoff$PID >= 94 & FWD49615_V4_BLAST_no_pid_cutoff$PID <= 97 , 1]
REV49615_readlitst_97_94 <- REV49615_V4_BLAST_no_pid_cutoff[ REV49615_V4_BLAST_no_pid_cutoff$PID >= 94 & REV49615_V4_BLAST_no_pid_cutoff$PID <= 97, 1]

#Give a histogram of the % matches to Silva (counting all hits, each read counted multiple times):    
FWD_Mapped_49615_97_94 <- filter(FWD_Mapped_49615_vs_Silva, Query %in% FWD49615_readlitst_97_94) %>%
  ggplot(aes(x=Percent_ID, fill=Genus)) +
    geom_histogram(breaks=seq(75,100, by = 1), color = "#e9ecef", alpha=0.6, position = 'identity') +
    scale_x_continuous(breaks = seq(75,100, by = 1)) +
    scale_fill_manual(values=c("#69b3a2", "#404080")) +
    ggtitle("Forward reads") +
    theme_classic() +
    theme(
      axis.line.x = element_line(size=0.1),
      axis.line.y = element_line(size=0.1),
      axis.title.x = element_text(size=12, margin = margin(t = 10, r = 0, b = 0, l = 0)), 
      axis.title.y = element_text(size=12, margin = margin(t = 0, r = 10, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 10, color = "black", angle = 45, vjust = 1, hjust = 1,
                                 margin = margin(t = 10, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 12, color = "black", margin = margin(t = 0, r = 10, b = 0, l = 0)),
      axis.ticks.length = unit(-0.1, "cm"),
      axis.ticks = element_line(size = 0.1),
      legend.position = "none",
      ) +
    ylab("Count") +
    xlab("Percent Match to Reference")

REV_Mapped_49615_97_94 <- filter(REV_Mapped_49615_vs_Silva, Query %in% REV49615_readlitst_97_94) %>%
  ggplot(aes(x=Percent_ID, fill=Genus)) +
    geom_histogram(breaks=seq(75,100, by = 1),  color = "#e9ecef", alpha=0.6, position = 'identity') +
    scale_x_continuous(breaks = seq(75,100, by = 1)) +
    scale_fill_manual(values=c("#69b3a2", "#404080")) +
    ggtitle("Reverse reads") +
    theme_classic() +
    theme(
      axis.line.x = element_line(size=0.1),
      axis.line.y = element_line(size=0.1),
      axis.title.x = element_text(size=12, margin = margin(t = 10, r = 0, b = 0, l = 0)), 
      axis.title.y = element_text(size=12, margin = margin(t = 0, r = 10, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 10, color = "black", angle = 45, vjust = 1, hjust = 1,
                                 margin = margin(t = 10, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 12, color = "black", margin = margin(t = 0, r = 10, b = 0, l = 0)),
      axis.ticks.length = unit(-0.1, "cm"),
      axis.ticks = element_line(size = 0.1)
      ) +
    ylab("Count") +
    xlab("Percent Match to Reference")

All_mapped_49615_97_94 <- FWD_Mapped_49615_97_94 + REV_Mapped_49615_97_94 + guide_area()
All_mapped_49615_97_94
ggsave("All_mapped_49615_97_94.pdf", plot = All_mapped_49615_97_94, width = 40, height = 18, units = "cm", dpi=320)
```
Make another histogram of percent match to Silva, but for reads that mapped to Microcystis at 100-95 percent identity:  
```{r}
#Get a list of reads that mapped to the Microcystis V4 at >= 95% ID:  
FWD49615_readlitst_100_95 <- FWD49615_V4_BLAST_no_pid_cutoff[ FWD49615_V4_BLAST_no_pid_cutoff$PID >= 95, 1]
REV49615_readlitst_100_95 <- REV49615_V4_BLAST_no_pid_cutoff[ REV49615_V4_BLAST_no_pid_cutoff$PID >= 95, 1]

#Give a histogram of the % matches to Silva (counting all hits, each read counted multiple times):    
FWD_Mapped_49615_100_95 <- filter(FWD_Mapped_49615_vs_Silva, Query %in% FWD49615_readlitst_100_95) %>%
  ggplot(aes(x=Percent_ID, fill=Genus)) +
    geom_histogram(breaks=seq(75,100, by = 1), color = "#e9ecef", alpha=0.6, position = 'identity') +
    scale_x_continuous(breaks = seq(75,100, by = 1)) +
    scale_fill_manual(values=c("#69b3a2", "#404080")) +
    ggtitle("Forward reads") +
    theme_classic() +
    theme(
      axis.line.x = element_line(size=0.1),
      axis.line.y = element_line(size=0.1),
      axis.title.x = element_text(size=12, margin = margin(t = 10, r = 0, b = 0, l = 0)), 
      axis.title.y = element_text(size=12, margin = margin(t = 0, r = 10, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 10, color = "black", angle = 45, vjust = 1, hjust = 1,
                                 margin = margin(t = 10, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 12, color = "black", margin = margin(t = 0, r = 10, b = 0, l = 0)),
      axis.ticks.length = unit(-0.1, "cm"),
      axis.ticks = element_line(size = 0.1),
      legend.position = "none",
      ) +
    ylab("Count") +
    xlab("Percent Match to Reference")

REV_Mapped_49615_100_95 <- filter(REV_Mapped_49615_vs_Silva, Query %in% REV49615_readlitst_100_95) %>%
  ggplot(aes(x=Percent_ID, fill=Genus)) +
    geom_histogram(breaks=seq(75,100, by = 1),  color = "#e9ecef", alpha=0.6, position = 'identity') +
    scale_x_continuous(breaks = seq(75,100, by = 1)) +
    scale_fill_manual(values=c("#69b3a2", "#404080")) +
    ggtitle("Reverse reads") +
    theme_classic() +
    theme(
      axis.line.x = element_line(size=0.1),
      axis.line.y = element_line(size=0.1),
      axis.title.x = element_text(size=12, margin = margin(t = 10, r = 0, b = 0, l = 0)), 
      axis.title.y = element_text(size=12, margin = margin(t = 0, r = 10, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 10, color = "black", angle = 45, vjust = 1, hjust = 1,
                                 margin = margin(t = 10, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 12, color = "black", margin = margin(t = 0, r = 10, b = 0, l = 0)),
      axis.ticks.length = unit(-0.1, "cm"),
      axis.ticks = element_line(size = 0.1)
      ) +
    ylab("Count") +
    xlab("Percent Match to Reference")

All_mapped_49615_100_95 <- FWD_Mapped_49615_100_95 + REV_Mapped_49615_100_95 + guide_area()
All_mapped_49615_100_95
ggsave("All_mapped_49615_100_95.pdf", plot = All_mapped_49615_100_95, width = 40, height = 18, units = "cm", dpi=320)
```

Make another histogram of percent match to Silva, but for reads that mapped to Microcystis at 100-95 percent identity and only including top hits to Silva:  
```{r}
#Import the matches to Silva:
FWD_Mapped_49615_vs_Silva_tophits <- read.table("49615_fwd_mapped_Microcystis_V4_vs_Silva_v138.blastn.postblast.tophits", sep="\t", header=FALSE)
REV_Mapped_49615_vs_Silva_tophits <- read.table("49615_rev_mapped_Microcystis_V4_vs_Silva_v138.blastn.postblast.tophits", sep="\t", header=FALSE)

colnames(FWD_Mapped_49615_vs_Silva_tophits) <- c("Query", "Subject", "Percent_ID", "Aln_Len", "Mismatch", "Gap", "Query_Start", "Query_Stop",
                                         "Subject_Start", "Subject_Stop", "E-Value", "Bit_Score", "Qlen", "Slen", "Qcovs")

colnames(REV_Mapped_49615_vs_Silva_tophits)<- c("Query", "Subject", "Percent_ID", "Aln_Len", "Mismatch", "Gap", "Query_Start", "Query_Stop",
                                         "Subject_Start", "Subject_Stop", "E-Value", "Bit_Score", "Qlen", "Slen", "Qcovs")

#Tidy up the taxonomy columns:  
FWD_Mapped_49615_vs_Silva_tophits <- separate(FWD_Mapped_49615_vs_Silva_tophits, Subject, c("Subject", "Domain", "Phylum", "Class", "Order",
                                                                                            "Family", "Genus"), sep=";", remove=TRUE)
REV_Mapped_49615_vs_Silva_tophits <- separate(REV_Mapped_49615_vs_Silva_tophits, Subject, c("Subject", "Domain", "Phylum", "Class", "Order",
                                                                                            "Family", "Genus"), sep=";", remove=TRUE)

#Make NAs in taxonomy columns be listed as unknown: 
FWD_Mapped_49615_vs_Silva_tophits[is.na(FWD_Mapped_49615_vs_Silva_tophits)] <- "unknown"
REV_Mapped_49615_vs_Silva_tophits[is.na(REV_Mapped_49615_vs_Silva_tophits)] <- "unknown"

#If the reference genus is not Microcystis, list it as "non_Microcystis" for plotting purposes:
FWD_Mapped_49615_vs_Silva_tophits$Genus[FWD_Mapped_49615_vs_Silva_tophits$Genus != "Microcystis_PCC7914"] <- "non_Microcystis"
REV_Mapped_49615_vs_Silva_tophits$Genus[REV_Mapped_49615_vs_Silva_tophits$Genus != "Microcystis_PCC7914"] <- "non_Microcystis"

#Remove all of the reads that matched to Silva with Read Coverage below 80% of read length:
FWD_Mapped_49615_vs_Silva_tophits <- FWD_Mapped_49615_vs_Silva_tophits[ FWD_Mapped_49615_vs_Silva_tophits$Qcovs >= 80 , ]
REV_Mapped_49615_vs_Silva_tophits <- REV_Mapped_49615_vs_Silva_tophits[ REV_Mapped_49615_vs_Silva_tophits$Qcovs >= 80, ]

#Give a histogram of the % matches to Silva (counting all hits, each read counted multiple times):    
FWD_Mapped_49615_100_95_tophits <- filter(FWD_Mapped_49615_vs_Silva_tophits, Query %in% FWD49615_readlitst_100_95) %>%
  ggplot(aes(x=Percent_ID, fill=Genus)) +
    geom_histogram(breaks=seq(75,100, by = 1), color = "#e9ecef", alpha=0.6, position = 'identity') +
    scale_x_continuous(breaks = seq(75,100, by = 1)) +
    scale_fill_manual(values=c("#69b3a2", "#404080")) +
    ggtitle("Forward reads") +
    theme_classic() +
    theme(
      axis.line.x = element_line(size=0.1),
      axis.line.y = element_line(size=0.1),
      axis.title.x = element_text(size=12, margin = margin(t = 10, r = 0, b = 0, l = 0)), 
      axis.title.y = element_text(size=12, margin = margin(t = 0, r = 10, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 10, color = "black", angle = 45, vjust = 1, hjust = 1,
                                 margin = margin(t = 10, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 12, color = "black", margin = margin(t = 0, r = 10, b = 0, l = 0)),
      axis.ticks.length = unit(-0.1, "cm"),
      axis.ticks = element_line(size = 0.1),
      legend.position = "none",
      ) +
    ylab("Count") +
    xlab("Percent Match to Reference")

REV_Mapped_49615_100_95_tophits <- filter(REV_Mapped_49615_vs_Silva_tophits, Query %in% REV49615_readlitst_100_95) %>%
  ggplot(aes(x=Percent_ID, fill=Genus)) +
    geom_histogram(breaks=seq(75,100, by = 1),  color = "#e9ecef", alpha=0.6, position = 'identity') +
    scale_x_continuous(breaks = seq(75,100, by = 1)) +
    scale_fill_manual(values=c("#69b3a2", "#404080")) +
    ggtitle("Reverse reads") +
    theme_classic() +
    theme(
      axis.line.x = element_line(size=0.1),
      axis.line.y = element_line(size=0.1),
      axis.title.x = element_text(size=12, margin = margin(t = 10, r = 0, b = 0, l = 0)), 
      axis.title.y = element_text(size=12, margin = margin(t = 0, r = 10, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 10, color = "black", angle = 45, vjust = 1, hjust = 1,
                                 margin = margin(t = 10, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 12, color = "black", margin = margin(t = 0, r = 10, b = 0, l = 0)),
      axis.ticks.length = unit(-0.1, "cm"),
      axis.ticks = element_line(size = 0.1)
      ) +
    ylab("Count") +
    xlab("Percent Match to Reference")

All_mapped_49615_100_95_tophits <- FWD_Mapped_49615_100_95_tophits + REV_Mapped_49615_100_95_tophits + guide_area()
All_mapped_49615_100_95_tophits
ggsave("All_mapped_49615_100_95_tophits.pdf", plot = All_mapped_49615_100_95_tophits, width = 40, height = 18, units = "cm", dpi=320)
```
How many of the "non-Microcystis" hits at 95-94% are to unclassified Microcystaecae?  
```{r}
#Make non-Microcystaceae hits listed generally as "non-Microcystaceae" for plotting purposes:
FWD_Mapped_49615_vs_Silva$Family[FWD_Mapped_49615_vs_Silva$Family != "Microcystaceae"] <- "non-Microcystaceae"
REV_Mapped_49615_vs_Silva$Family[REV_Mapped_49615_vs_Silva$Family != "Microcystaceae"] <- "non-Microcystaceae"

#Give a histogram of the % matches to Silva (counting all hits, each read counted multiple times):    
FWD_Mapped_49615_100_95_fam <- filter(FWD_Mapped_49615_vs_Silva, Query %in% FWD49615_readlitst_100_95) %>%
  ggplot(aes(x=Percent_ID, fill=Family)) +
    geom_histogram(breaks=seq(75,100, by = 1), color = "#e9ecef", alpha=0.6, position = 'identity') +
    scale_x_continuous(breaks = seq(75,100, by = 1)) +
    scale_fill_manual(values=c("#69b3a2", "#404080")) +
    ggtitle("Forward reads") +
    theme_classic() +
    theme(
      axis.line.x = element_line(size=0.1),
      axis.line.y = element_line(size=0.1),
      axis.title.x = element_text(size=12, margin = margin(t = 10, r = 0, b = 0, l = 0)), 
      axis.title.y = element_text(size=12, margin = margin(t = 0, r = 10, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 10, color = "black", angle = 45, vjust = 1, hjust = 1,
                                 margin = margin(t = 10, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 12, color = "black", margin = margin(t = 0, r = 10, b = 0, l = 0)),
      axis.ticks.length = unit(-0.1, "cm"),
      axis.ticks = element_line(size = 0.1),
      legend.position = "none",
      ) +
    ylab("Count") +
    xlab("Percent Match to Reference")

REV_Mapped_49615_100_95_fam <- filter(REV_Mapped_49615_vs_Silva, Query %in% REV49615_readlitst_100_95) %>%
  ggplot(aes(x=Percent_ID, fill=Family)) +
    geom_histogram(breaks=seq(75,100, by = 1),  color = "#e9ecef", alpha=0.6, position = 'identity') +
    scale_x_continuous(breaks = seq(75,100, by = 1)) +
    scale_fill_manual(values=c("#69b3a2", "#404080")) +
    ggtitle("Reverse reads") +
    theme_classic() +
    theme(
      axis.line.x = element_line(size=0.1),
      axis.line.y = element_line(size=0.1),
      axis.title.x = element_text(size=12, margin = margin(t = 10, r = 0, b = 0, l = 0)), 
      axis.title.y = element_text(size=12, margin = margin(t = 0, r = 10, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 10, color = "black", angle = 45, vjust = 1, hjust = 1,
                                 margin = margin(t = 10, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 12, color = "black", margin = margin(t = 0, r = 10, b = 0, l = 0)),
      axis.ticks.length = unit(-0.1, "cm"),
      axis.ticks = element_line(size = 0.1)
      ) +
    ylab("Count") +
    xlab("Percent Match to Reference")

All_mapped_49615_100_95_fam <- FWD_Mapped_49615_100_95_fam + REV_Mapped_49615_100_95_fam + guide_area()
All_mapped_49615_100_95_fam
ggsave("All_mapped_49615_100_95_fam.pdf", plot = All_mapped_49615_100_95_fam, width = 40, height = 18, units = "cm", dpi=320)
```
Make a histogram of bit score for the reads that mapped to the V4 sequence at 100-95% identity: 
```{r}
#Give a histogram of the % matches to Silva (counting all hits, each read counted multiple times):    
FWD_Mapped_49615_100_95_bit_score <- filter(FWD_Mapped_49615_vs_Silva, Query %in% FWD49615_readlitst_100_95) %>%
  ggplot(aes(x=Bit_Score, fill=Genus)) +
    geom_histogram(breaks=seq(160,230, by=2), color = "#e9ecef", alpha=0.6, position = 'identity') +
    scale_x_continuous(breaks = seq(160,230, by=4)) +
    scale_fill_manual(values=c("#69b3a2", "#404080")) +
    ggtitle("Forward reads, all hits") +
    theme_classic() +
    theme(
      axis.line.x = element_line(size=0.1),
      axis.line.y = element_line(size=0.1),
      axis.title.x = element_text(size=12, margin = margin(t = 10, r = 0, b = 0, l = 0)), 
      axis.title.y = element_text(size=12, margin = margin(t = 0, r = 10, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 9, color = "black", angle = 45, vjust = 1, hjust = 1,
                                 margin = margin(t = 10, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 12, color = "black", margin = margin(t = 0, r = 10, b = 0, l = 0)),
      axis.ticks.length = unit(-0.1, "cm"),
      axis.ticks = element_line(size = 0.1),
      legend.position = "none",
      ) +
    ylab("Count") +
    xlab("Bit Score")

REV_Mapped_49615_100_95_bit_score <- filter(REV_Mapped_49615_vs_Silva, Query %in% REV49615_readlitst_100_95) %>%
  ggplot(aes(x=Bit_Score, fill=Genus)) +
    geom_histogram(breaks=seq(160,230, by=2), color = "#e9ecef", alpha=0.6, position = 'identity') +
    scale_x_continuous(breaks = seq(160,230, by=4)) +
    scale_fill_manual(values=c("#69b3a2", "#404080")) +
    ggtitle("Reverse reads, all hits") +
    theme_classic() +
    theme(
      axis.line.x = element_line(size=0.1),
      axis.line.y = element_line(size=0.1),
      axis.title.x = element_text(size=12, margin = margin(t = 10, r = 0, b = 0, l = 0)), 
      axis.title.y = element_text(size=12, margin = margin(t = 0, r = 10, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 9, color = "black", angle = 45, vjust = 1, hjust = 1,
                                 margin = margin(t = 10, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 12, color = "black", margin = margin(t = 0, r = 10, b = 0, l = 0)),
      axis.ticks.length = unit(-0.1, "cm"),
      axis.ticks = element_line(size = 0.1)
      ) +
    ylab("Count") +
    xlab("Bit Score")

#Make a histogram of bit score for just the top hits to Silva:  
FWD_Mapped_49615_100_95_tophits_bitscore <- filter(FWD_Mapped_49615_vs_Silva_tophits, Query %in% FWD49615_readlitst_100_95) %>%
  ggplot(aes(x=Bit_Score, fill=Genus)) +
    geom_histogram(breaks=seq(160,230, by=2), color = "#e9ecef", alpha=0.6, position = 'identity') +
    scale_x_continuous(breaks = seq(160,230, by=4)) +
    scale_fill_manual(values=c("#69b3a2", "#404080")) +
    ggtitle("Forward reads, best hits only") +
    theme_classic() +
    theme(
      axis.line.x = element_line(size=0.1),
      axis.line.y = element_line(size=0.1),
      axis.title.x = element_text(size=12, margin = margin(t = 10, r = 0, b = 0, l = 0)), 
      axis.title.y = element_text(size=12, margin = margin(t = 0, r = 10, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 9, color = "black", angle = 45, vjust = 1, hjust = 1,
                                 margin = margin(t = 10, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 12, color = "black", margin = margin(t = 0, r = 10, b = 0, l = 0)),
      axis.ticks.length = unit(-0.1, "cm"),
      axis.ticks = element_line(size = 0.1),
      legend.position = "none",
      ) +
    ylab("Count") +
    xlab("Bit Score")

REV_Mapped_49615_100_95_tophits_bitscore <- filter(REV_Mapped_49615_vs_Silva_tophits, Query %in% REV49615_readlitst_100_95) %>%
  ggplot(aes(x=Bit_Score, fill=Genus)) +
    geom_histogram(breaks=seq(160,230, by=2),  color = "#e9ecef", alpha=0.6, position = 'identity') +
    scale_x_continuous(breaks = seq(160,230, by=4)) +
    scale_fill_manual(values=c("#69b3a2", "#404080")) +
    ggtitle("Reverse reads, best hits only") +
    theme_classic() +
    theme(
      axis.line.x = element_line(size=0.1),
      axis.line.y = element_line(size=0.1),
      axis.title.x = element_text(size=12, margin = margin(t = 10, r = 0, b = 0, l = 0)), 
      axis.title.y = element_text(size=12, margin = margin(t = 0, r = 10, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 9, color = "black", angle = 45, vjust = 1, hjust = 1,
                                 margin = margin(t = 10, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 12, color = "black", margin = margin(t = 0, r = 10, b = 0, l = 0)),
      axis.ticks.length = unit(-0.1, "cm"),
      axis.ticks = element_line(size = 0.1),
      legend.position = "none"
      ) +
    ylab("Count") +
    xlab("Bit Score")

All_mapped_49615_100_95_bit_score <- FWD_Mapped_49615_100_95_bit_score + REV_Mapped_49615_100_95_bit_score + guide_area() + FWD_Mapped_49615_100_95_tophits_bitscore + REV_Mapped_49615_100_95_tophits_bitscore + plot_layout(ncol=3)

All_mapped_49615_100_95_bit_score
ggsave("All_mapped_49615_100_95_bit_score.pdf", plot = All_mapped_49615_100_95_bit_score, width = 40, height = 18, units = "cm", dpi=320)
```
What proportion of the Microcystaceae hits in Silva are not from Microcystis?  
```{r}
#Import the matches to Silva again so that we can keep the taxonomy (it was edited above earlier for easier grouping):
FWD_Mapped_49615_vs_Silva_with_Genus <- read.table("49615_fwd_mapped_Microcystis_V4_vs_Silva_v138.blastn.postblast", sep="\t", header=TRUE)
REV_Mapped_49615_vs_Silva_with_Genus <- read.table("49615_rev_mapped_Microcystis_V4_vs_Silva_v138.blastn.postblast", sep="\t", header=TRUE)

#Remove all of the reads that matched to Silva with Read Coverage below 80% of read length:
FWD_Mapped_49615_vs_Silva_with_Genus <- FWD_Mapped_49615_vs_Silva_with_Genus[ FWD_Mapped_49615_vs_Silva_with_Genus$Qcovs >= 80 , ]
REV_Mapped_49615_vs_Silva_with_Genus <- REV_Mapped_49615_vs_Silva_with_Genus[ REV_Mapped_49615_vs_Silva_with_Genus$Qcovs >= 80 , ]

#Keep only reads that mapped to the Microcystis V4 at 100-95% identity:
FWD_Mapped_49615_vs_Silva_with_Genus <- filter(FWD_Mapped_49615_vs_Silva_with_Genus, Query %in% FWD49615_readlitst_100_95) 
REV_Mapped_49615_vs_Silva_with_Genus <- filter(REV_Mapped_49615_vs_Silva_with_Genus, Query %in% REV49615_readlitst_100_95)

#Tidy up the taxonomy columns:  
FWD_Mapped_49615_vs_Silva_with_Genus <- separate(FWD_Mapped_49615_vs_Silva_with_Genus, Subject, c("Subject", "Domain", "Phylum", "Class", "Order", "Family", "Genus"), sep=";", remove=TRUE)

REV_Mapped_49615_vs_Silva_with_Genus <- separate(REV_Mapped_49615_vs_Silva_with_Genus, Subject, c("Subject", "Domain", "Phylum", "Class", "Order", "Family", "Genus"), sep=";", remove=TRUE)

#Make NAs in taxonomy columns be listed as unknown: 
FWD_Mapped_49615_vs_Silva_with_Genus[is.na(FWD_Mapped_49615_vs_Silva_with_Genus)] <- "unknown"
REV_Mapped_49615_vs_Silva_with_Genus[is.na(REV_Mapped_49615_vs_Silva_with_Genus)] <- "unknown"

#Combine the forward and reverse hits into a single table:
Total_Mapped_49615_vs_Silva_with_Genus <- rbind(FWD_Mapped_49615_vs_Silva_with_Genus, REV_Mapped_49615_vs_Silva_with_Genus)

#Tally the reads by Taxonomy, but only include Microcystaceae hits:
Total_Mapped_49615_Microcystaceae_summary <- filter(Total_Mapped_49615_vs_Silva_with_Genus, Family == "Microcystaceae") %>%
  group_by(Genus) %>%
  summarise(Counts=n(), avg_pid=mean(Percent_ID), sd_pid=sd(Percent_ID)) %>%
  mutate(CI95=sd_pid/sqrt(Counts)*1.96)

#Make a rank abundance plot of the Microcystaceae hits:
Total_Mapped_49615_100_95_Silva_Tax <- Total_Mapped_49615_Microcystaceae_summary %>%
                        arrange(desc(Counts)) %>%
                            ggplot(aes(x=reorder(Genus, -Counts), y=Counts)) +
                              geom_bar(stat = "identity", position = position_dodge()) +
                              theme_classic() +
                              theme(
                                axis.line.x = element_line(size=0.1),
                                axis.line.y = element_line(size=0.1),
                                axis.title.x = element_text(size=12, margin = margin(t = 10, r = 0, b = 0, l = 0)), 
                                axis.title.y = element_text(size=12, margin = margin(t = 0, r = 10, b = 0, l = 0)),
                                panel.grid.major = element_blank(),
                                panel.grid.minor = element_blank(),
                                axis.text.x = element_text(size = 10, color = "black", angle = 45, vjust = 1, hjust = 1,
                                                           margin = margin(t = 10, r = 0, b = 0, l = 0)),
                                axis.text.y = element_text(size = 10, color = "black", margin = margin(t = 0, r = 10, b = 0, l = 0)),
                                axis.ticks.length = unit(-0.1, "cm"),
                                axis.ticks = element_line(size = 0.1)
                                ) +
                              ylab("Total reads mapped") +
                              xlab("Taxon") 

Total_Mapped_49615_100_95_Silva_Tax
ggsave("Total_Mapped_49615_100_95_Silva_Tax.pdf", plot = Total_Mapped_49615_100_95_Silva_Tax, width = 22, height = 18, units = "cm", dpi=320)
```
Make a boxplot of percent identity for each other genus:  
```{r}
Total_Mapped_49615_100_95_Silva_PID <- filter(Total_Mapped_49615_vs_Silva_with_Genus, Family == "Microcystaceae") %>%
  ggplot(aes(x=Genus, y=Percent_ID)) +
    geom_boxplot() +
    theme_classic() +
    theme(
      axis.line.x = element_line(size=0.1),
      axis.line.y = element_line(size=0.1),
      axis.title.x = element_text(size=12, margin = margin(t = 10, r = 0, b = 0, l = 0)), 
      axis.title.y = element_text(size=12, margin = margin(t = 0, r = 10, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 10, color = "black", angle = 45, vjust = 1, hjust = 1,
                                 margin = margin(t = 10, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 10, color = "black", margin = margin(t = 0, r = 10, b = 0, l = 0)),
      axis.ticks.length = unit(-0.1, "cm"),
      axis.ticks = element_line(size = 0.1)
      ) +
    ylab("Percent Identity to Silva") +
    xlab("Taxon")

Total_Mapped_49615_100_95_Silva_PID
ggsave("Total_Mapped_49615_100_95_Silva_PID.pdf", plot = Total_Mapped_49615_100_95_Silva_PID, width = 22, height = 18, units = "cm", dpi=320)
```

Next BLAST the reads from sample 49615 that mapped to the mcy operon in Microcystis at 100-95% identity against NCBI nucleotide database to see if the results with diamond and muscato to teal's URDB are comparable to those obtained with blastn.  
```{r}
#Import the raw blast dataframe and count dataframe:
FWD49615_vs_NCBI_raw_blast <- read.table("49615_fwd_mapped_mcy_100_95_vs_nt.blastn.postblast.trimmed", sep="\t", header=FALSE)

colnames(FWD49615_vs_NCBI_raw_blast) <- c("Query", "Subject_gi", "Subject_acc", "PID", "Aln_Len", "Mismatch",
                                          "Gaps", "Qstart", "QStop", "Sstart", "Sstop", "Eval", "Bitscore",
                                          "Qlen", "Slen", "Qcovs")

REV49615_vs_NCBI_raw_blast <- read.table("49615_rev_mapped_mcy_100_95_vs_nt.blastn.postblast.trimmed", sep="\t", header=FALSE)

colnames(REV49615_vs_NCBI_raw_blast) <- c("Query", "Subject_gi", "Subject_acc", "PID", "Aln_Len", "Mismatch",
                                          "Gaps", "Qstart", "QStop", "Sstart", "Sstop", "Eval", "Bitscore",
                                          "Qlen", "Slen", "Qcovs")

FWD49615_vs_NCBI_counts <- read.table("Read_counts_49615_fwd_mapped_mcy_100_95_vs_nt.blastn.postblast.trimmed", sep="\t", header=FALSE)

colnames(FWD49615_vs_NCBI_counts) <- c("Subject_acc", "FWD_count", "Slen", "Annotation")

REV49615_vs_NCBI_counts <- read.table("Read_counts_49615_rev_mapped_mcy_100_95_vs_nt.blastn.postblast.trimmed", sep="\t", header=FALSE)

colnames(REV49615_vs_NCBI_counts) <- c("Subject_acc", "REV_count", "Slen", "Annotation")

#Combined the FWD and REV read counts:  
Combined49615_vs_NCBI_counts <- merge(FWD49615_vs_NCBI_counts, REV49615_vs_NCBI_counts,
                                      by = c("Subject_acc", "Slen", "Annotation"), all = TRUE)

#Turn all NAs in the count columns to zeros: 
Combined49615_vs_NCBI_counts$FWD_count[is.na(Combined49615_vs_NCBI_counts$FWD_count)] <- 0
Combined49615_vs_NCBI_counts$REV_count[is.na(Combined49615_vs_NCBI_counts$REV_count)] <- 0

#Sum the forward and reverse read hits:
Combined49615_vs_NCBI_counts$Total_counts <- Combined49615_vs_NCBI_counts$FWD_count + Combined49615_vs_NCBI_counts$REV_count

#Remove entries that did not have hits that met the BLAST filtering cutoffs (alignment coverage greater than 80% of read length):
Combined49615_vs_NCBI_counts <- filter(Combined49615_vs_NCBI_counts, Total_counts > 0)

#Make a column of cleaned up Genus names for plotting:
Combined49615_vs_NCBI_counts$Genus <- gsub(".*Microcystis.*", "Microcystis",
                                           Combined49615_vs_NCBI_counts$Annotation, ignore.case = TRUE)

Combined49615_vs_NCBI_counts$Genus <- gsub(".*Anabaena.*", "Anabaena",
                                           Combined49615_vs_NCBI_counts$Genus, ignore.case = TRUE)

Combined49615_vs_NCBI_counts$Genus <- gsub(".*Bacillus.*", "Bacillus",
                                           Combined49615_vs_NCBI_counts$Genus, ignore.case = TRUE)

Combined49615_vs_NCBI_counts$Genus <- gsub(".*Anabaenopsis.*", "Anabaenopsis",
                                           Combined49615_vs_NCBI_counts$Genus, ignore.case = TRUE)

Combined49615_vs_NCBI_counts$Genus <- gsub(".*Calothrix.*", "Calothrix",
                                           Combined49615_vs_NCBI_counts$Genus, ignore.case = TRUE)

Combined49615_vs_NCBI_counts$Genus <- gsub(".*Chondrocystis.*", "Chondrocystis",
                                           Combined49615_vs_NCBI_counts$Genus, ignore.case = TRUE)

Combined49615_vs_NCBI_counts$Genus <- gsub(".*Chroococcidiopsis.*", "Chroococcidiopsis",
                                           Combined49615_vs_NCBI_counts$Genus, ignore.case = TRUE)

Combined49615_vs_NCBI_counts$Genus <- gsub(".*Crinalium.*", "Crinalium",
                                           Combined49615_vs_NCBI_counts$Genus, ignore.case = TRUE)

Combined49615_vs_NCBI_counts$Genus <- gsub(".*Cyanothece.*", "Cyanothece",
                                           Combined49615_vs_NCBI_counts$Genus, ignore.case = TRUE)

Combined49615_vs_NCBI_counts$Genus <- gsub(".*Cylindrospermopsis.*", "Cylindrospermopsis",
                                           Combined49615_vs_NCBI_counts$Genus, ignore.case = TRUE)

Combined49615_vs_NCBI_counts$Genus <- gsub(".*Cylindrospermum.*", "Cylindrospermum",
                                           Combined49615_vs_NCBI_counts$Genus, ignore.case = TRUE)

Combined49615_vs_NCBI_counts$Genus <- gsub(".*Fischerella.*", "Fischerella",
                                           Combined49615_vs_NCBI_counts$Genus, ignore.case = TRUE)

	Combined49615_vs_NCBI_counts$Genus <- gsub(".*Fremyella.*", "Fremyella",
                                           Combined49615_vs_NCBI_counts$Genus, ignore.case = TRUE)

Combined49615_vs_NCBI_counts$Genus <- gsub(".*Gloeocapsa.*", "Gloeocapsa",
                                           Combined49615_vs_NCBI_counts$Genus, ignore.case = TRUE)

Combined49615_vs_NCBI_counts$Genus <- gsub(".*Hapalosiphon.*", "Hapalosiphon",
                                           Combined49615_vs_NCBI_counts$Genus, ignore.case = TRUE)

Combined49615_vs_NCBI_counts$Genus <- gsub(".*Legionella.*", "Legionella",
                                           Combined49615_vs_NCBI_counts$Genus, ignore.case = TRUE)

Combined49615_vs_NCBI_counts$Genus <- gsub(".*Leptolyngbya.*", "Leptolyngbya",
                                           Combined49615_vs_NCBI_counts$Genus, ignore.case = TRUE)

Combined49615_vs_NCBI_counts$Genus <- gsub(".*Lysinibacillus.*", "Lysinibacillus",
                                           Combined49615_vs_NCBI_counts$Genus, ignore.case = TRUE)

Combined49615_vs_NCBI_counts$Genus <- gsub(".*Moorea.*", "Moorea",
                                           Combined49615_vs_NCBI_counts$Genus, ignore.case = TRUE)

Combined49615_vs_NCBI_counts$Genus <- gsub(".*Nodularia.*", "Nodularia",
                                           Combined49615_vs_NCBI_counts$Genus, ignore.case = TRUE)

Combined49615_vs_NCBI_counts$Genus <- gsub(".*Nostoc.*", "Nostoc",
                                           Combined49615_vs_NCBI_counts$Genus, ignore.case = TRUE)

Combined49615_vs_NCBI_counts$Genus <- gsub(".*Oscillatoria.*", "Oscillatoria",
                                           Combined49615_vs_NCBI_counts$Genus, ignore.case = TRUE)

Combined49615_vs_NCBI_counts$Genus <- gsub(".*Phormidium.*", "Phormidium",
                                           Combined49615_vs_NCBI_counts$Genus, ignore.case = TRUE)

Combined49615_vs_NCBI_counts$Genus <- gsub(".*Planktothrix.*", "Planktothrix",
                                           Combined49615_vs_NCBI_counts$Genus, ignore.case = TRUE)

Combined49615_vs_NCBI_counts$Genus <- gsub(".*Rivularia.*", "Rivularia",
                                           Combined49615_vs_NCBI_counts$Genus, ignore.case = TRUE)

Combined49615_vs_NCBI_counts$Genus <- gsub(".*Scytonema.*", "Scytonema",
                                           Combined49615_vs_NCBI_counts$Genus, ignore.case = TRUE)

Combined49615_vs_NCBI_counts$Genus <- gsub(".*Stanieria.*", "Stanieria",
                                           Combined49615_vs_NCBI_counts$Genus, ignore.case = TRUE)

Combined49615_vs_NCBI_counts$Genus <- gsub(".*Synechococcus.*", "Synechococcus",
                                           Combined49615_vs_NCBI_counts$Genus, ignore.case = TRUE)

Combined49615_vs_NCBI_counts$Genus <- gsub(".*Tolypothrix.*", "Tolypothrix",
                                           Combined49615_vs_NCBI_counts$Genus, ignore.case = TRUE)

Combined49615_vs_NCBI_counts$Genus <- gsub(".*Uncultured cyanobacterium.*", "Uncultured cyanobacterium",
                                           Combined49615_vs_NCBI_counts$Genus, ignore.case = TRUE)

Combined49615_vs_NCBI_counts$Genus <- gsub(".*Uncultured bacterium.*", "Uncultured bacterium",
                                           Combined49615_vs_NCBI_counts$Genus, ignore.case = TRUE)

Combined49615_vs_NCBI_counts$Genus <- gsub(".*Cyanobacterium sp.*", "Cyanobacterium sp.",
                                           Combined49615_vs_NCBI_counts$Genus, ignore.case = TRUE)
```

Make a ranked abundance plot of hits summed by genus:  
```{r}
#Make a summary of the total counts by genus:
Combined49615_vs_NCBI_counts_Genus_summary <- Combined49615_vs_NCBI_counts %>%
  group_by(Genus) %>%
  summarise(Total=sum(Total_counts))

#Plot the summary:
Combined49615_vs_NCBI_counts_by_Genus <- Combined49615_vs_NCBI_counts_Genus_summary %>%
                        arrange(desc(Total)) %>%
                            ggplot(aes(x=reorder(Genus, -Total), y=Total)) +
                              geom_bar(stat = "identity", position = position_dodge()) +
                              theme_classic() +
                              theme(
                                axis.line.x = element_line(size=0.1),
                                axis.line.y = element_line(size=0.1),
                                axis.title.x = element_text(size=12, margin = margin(t = 10, r = 0, b = 0, l = 0)), 
                                axis.title.y = element_text(size=12, margin = margin(t = 0, r = 10, b = 0, l = 0)),
                                panel.grid.major = element_blank(),
                                panel.grid.minor = element_blank(),
                                axis.text.x = element_text(size = 10, color = "black", angle = 45, vjust = 1, hjust = 1, margin = margin(t = 10, r = 0, b = 0, l = 0)),
                                axis.text.y = element_text(size = 10, color = "black", margin = margin(t = 0, r = 10, b = 0, l = 0)),
                                axis.ticks.length = unit(-0.1, "cm"),
                                axis.ticks = element_line(size = 0.1)
                                ) +
                              ylab("Total reads mapped") +
                              xlab("Taxon") 

Combined49615_vs_NCBI_counts_by_Genus
ggsave("Combined49615_vs_NCBI_counts_by_Genus.pdf", plot = Combined49615_vs_NCBI_counts_by_Genus, width = 22, height = 18, units = "cm", dpi=320)
```
Most of the hits to NCBI nr are to Microcystis (this includes redundant hits, reads were counted multiple times).  

What is the average percent identity of the non-Microcystis hits in NCBI nr?  
```{r}
#concatenate the rows of the forward and reverse raw BLAST output:
Combined49615_vs_NCBI_raw_blast <- rbind(FWD49615_vs_NCBI_raw_blast, REV49615_vs_NCBI_raw_blast)

#Merge the Annotation counts dataframe with the blast result dataframe so that I can group the individual hits by genus:  
Annotated_BLAST_output <- merge(Combined49615_vs_NCBI_raw_blast, Combined49615_vs_NCBI_counts, all.x = TRUE, all.y = FALSE, by = c("Subject_acc", "Slen"))

#Remove the count columns, which don't make sense with this dataframe set up:
drop <- c("FWD_count", "REV_count", "Total_counts")
Annotated_BLAST_output <- Annotated_BLAST_output[ , !(colnames(Annotated_BLAST_output) %in% drop)]

#Remove the hits where the query coverage is less than 80%:
Annotated_BLAST_output <- filter(Annotated_BLAST_output, Qcovs >= 80)

#Make a boxplot of hit percent identity to NCBI nr (grouped by Genus):
Mcy_Mapped_49615_100_95_NCBI_PID <- ggplot(Annotated_BLAST_output, aes(x=Genus, y=PID)) +
    geom_boxplot() +
    theme_classic() +
    theme(
      axis.line.x = element_line(size=0.1),
      axis.line.y = element_line(size=0.1),
      axis.title.x = element_text(size=12, margin = margin(t = 10, r = 0, b = 0, l = 0)), 
      axis.title.y = element_text(size=12, margin = margin(t = 0, r = 10, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 10, color = "black", angle = 45, vjust = 1, hjust = 1,
                                 margin = margin(t = 10, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 10, color = "black", margin = margin(t = 0, r = 10, b = 0, l = 0)),
      axis.ticks.length = unit(-0.1, "cm"),
      axis.ticks = element_line(size = 0.1)
      ) +
    ylab("Percent Identity to NCBI NT") +
    xlab("Taxon")

Mcy_Mapped_49615_100_95_NCBI_PID
ggsave("Mcy_Mapped_49615_100_95_NCBI_PID.pdf", plot = Mcy_Mapped_49615_100_95_NCBI_PID, width = 22, height = 18, units = "cm", dpi=320)
```

What is the media PID that matched to Microcystis?
```{r}
MC_mcy_hits_only <- filter(Annotated_BLAST_output, Genus == "Microcystis")
quantile(MC_mcy_hits_only$PID)
```

Create a histogram of Microcystis and non-Microcystis bit scores for the mcy reads:  
```{r}
#Create a dataframe that has Genus as a binary Microcystis or non-Microcystis:
Annotated_BLAST_output_binary_genus <- Annotated_BLAST_output
Annotated_BLAST_output_binary_genus$Genus[Annotated_BLAST_output_binary_genus$Genus != "Microcystis"] <- "non-Microcystis"

#Give a histogram of the % matches to Silva (counting all hits, each read counted multiple times):    
Combined49615_100_95_mcy_mapped_bit_score <- ggplot(Annotated_BLAST_output_binary_genus, aes(x=Bitscore, fill=Genus)) +
    geom_histogram(breaks = seq(50,230, by=4), color = "#e9ecef", alpha=0.6, position = 'identity') +
    scale_x_continuous(breaks = seq(50,230, by=4)) +
    scale_fill_manual(values=c("#69b3a2", "#404080")) +
    theme_classic() +
    theme(
      axis.line.x = element_line(size=0.1),
      axis.line.y = element_line(size=0.1),
      axis.title.x = element_text(size=12, margin = margin(t = 10, r = 0, b = 0, l = 0)), 
      axis.title.y = element_text(size=12, margin = margin(t = 0, r = 10, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 9, color = "black", angle = 45, vjust = 1, hjust = 1,
                                 margin = margin(t = 10, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 12, color = "black", margin = margin(t = 0, r = 10, b = 0, l = 0)),
      axis.ticks.length = unit(-0.1, "cm"),
      axis.ticks = element_line(size = 0.1),
      legend.position = "top"
      ) +
    ylab("Count") +
    xlab("Bit Score")


Combined49615_100_95_mcy_mapped_bit_score
ggsave("Combined49615_100_95_mcy_mapped_bit_score.pdf", plot = Combined49615_100_95_mcy_mapped_bit_score, width = 22, height = 18, units = "cm", dpi=320)
```
Remake the above bit score histogram, removing the hits to uncultured clones which may be Microcystis.  
Confirmed by running the sequences through the BLAST webtool, many had hits at 99% identity and 100% query coverage to Microcystis microcystin synthetase genes.  
```{r}
No_uncultured_mcy_mapped_bit_score <- filter(Annotated_BLAST_output_binary_genus, Genus != "Uncultured cyanobacterium" & Genus != "Uncultured bacterium") %>% 
  ggplot(aes(x=Bitscore, fill=Genus)) +
    geom_histogram(breaks = seq(50,230, by=4), color = "#e9ecef", alpha=0.6, position = 'identity') +
    scale_x_continuous(breaks = seq(50,230, by=4)) +
    scale_fill_manual(values=c("#69b3a2", "#404080")) +
    theme_classic() +
    theme(
      axis.line.x = element_line(size=0.1),
      axis.line.y = element_line(size=0.1),
      axis.title.x = element_text(size=12, margin = margin(t = 10, r = 0, b = 0, l = 0)), 
      axis.title.y = element_text(size=12, margin = margin(t = 0, r = 10, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 9, color = "black", angle = 45, vjust = 1, hjust = 1,
                                 margin = margin(t = 10, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 12, color = "black", margin = margin(t = 0, r = 10, b = 0, l = 0)),
      axis.ticks.length = unit(-0.1, "cm"),
      axis.ticks = element_line(size = 0.1),
      legend.position = "top"
      ) +
    ylab("Count") +
    xlab("Bit Score")

No_uncultured_mcy_mapped_bit_score
ggsave("No_uncultured_mcy_mapped_bit_score.pdf", plot = No_uncultured_mcy_mapped_bit_score, width = 22, height = 18, units = "cm", dpi=320)
```
We want to include all of the entries in NCBI that recruited reads that are not from Microcystis in the BLAST databases as a reference so that the BLAST is more competitive, so I need to get a list of accession numbers that are only for the non-Microcystis hits:  
```{r}
#Remove the Microcystis hits and unculture hits (which look like they are mostly microcystis) from the Blast dataframe:
Combined49615_vs_NCBI_counts_no_Microcystis <- filter(Combined49615_vs_NCBI_counts, Genus != "Microcystis" & Genus != "Uncultured bacterium" & Genus != "Uncultured cyanobacterium")

#Save the table:
write.table(Combined49615_vs_NCBI_counts_no_Microcystis, file = "Combined49615_vs_NCBI_counts_no_Microcystis.txt", sep="\t")
```

The reads for all samples were BLAST against more competitive databases. The 16S rRNA database was all of the cyanobacteria sequences included the V4 region database created in mothur SOPs from the Silva v138 reference database. The microcystin database included the complete operon from Microcystis genomes available in IMG, along with non-Microcystis sequences from NCBI that recruited reads that mapped to the Microcystin operon above. The percent identity filter for both BLASTs was 95%, the alignment length cutoff was 80% of read length. Reads were counted based on the bitscore, reads with an obvious best hit were only counted once.

Robert wrote a script that counts ambiguous hits in three different ways. Ambiguous hits are reads that map to both Microcystis and a non-Microcystis at equal bit score. The script would either 1) double count all ambiguous hits, 2) remove all ambiguous hits, or 3) divide the count of hits by the number of matches that it went to.

Inspecting the output of the script below:  
```{r}
No_ambig_counts <- read.table("Read_couts_mcy_cyano16S.aln_len_filter.none.csv", sep="\t", header= TRUE)
All_ambig_counts <- read.table("Read_couts_mcy_cyano16S.aln_len_filter.all.csv", sep="\t", header= TRUE)
Divide_ambig_counts <- read.table("Read_couts_mcy_cyano16S.aln_len_filter.part.csv", sep="\t", header= TRUE)
```

Calculate % toxicity for each sample using each counting method:  
```{r}
No_ambig_counts_Microcystis_16S_counts <- No_ambig_counts[ No_ambig_counts$X == "Microcystis", ]
All_ambig_counts_Microcystis_16S_counts <- All_ambig_counts[ All_ambig_counts$X == "Microcystis", ]
Divide_ambig_counts_Microcystis_16S_counts <- Divide_ambig_counts[ Divide_ambig_counts$X == "Microcystis", ]

No_ambig_counts_Microcystis_mcyD_counts <- No_ambig_counts[ No_ambig_counts$X == "Microcystis_McyD", ]
All_ambig_counts_Microcystis_mcyD_counts <- All_ambig_counts[ All_ambig_counts$X == "Microcystis_McyD", ]
Divide_ambig_counts_Microcystis_mcyD_counts <- Divide_ambig_counts[ Divide_ambig_counts$X == "Microcystis_McyD", ]

#create an empty array to store calculations in:
Percent_Toxic_Microcystis <- array(numeric(), c(45,7))

#create a vector of file names to loop through:
Files <- colnames(No_ambig_counts_Microcystis)
drop <- "X"
Files <- Files[!(Files %in%drop)]

#Create a vector of sample names to loop through:
Samples <- gsub("_adtrim_clean_qtrim_.*", "", colnames(No_ambig_counts_Microcystis)[2:61], perl=TRUE)
Samples <- unique(Samples)

## Create the table:
#Set counters to starting values
FWD_16S_counter <- 1
FWD_mcy_counter <- 2
REV_16S_counter <- 3
REV_mcy_counter <- 4
#These counters will track the new array positions that are filled to prevent data overwrite:
row_tracker <- 1
#Loop:
for (Sample in Samples){
  #This will generate the data for the no ambig counts
  #Sample ID
  Percent_Toxic_Microcystis[row_tracker, 1] <- Sample
  #Count Method
  Percent_Toxic_Microcystis[row_tracker, 2] <- "No ambig"
  #Total McyD counts
  Percent_Toxic_Microcystis[row_tracker, 3] <- No_ambig_counts_Microcystis_mcyD_counts[,Files[FWD_mcy_counter]] + No_ambig_counts_Microcystis_mcyD_counts[,Files[REV_mcy_counter]]
  #Total 16S V4 counts
  Percent_Toxic_Microcystis[row_tracker, 4] <- No_ambig_counts_Microcystis_16S_counts[,Files[FWD_16S_counter]] + No_ambig_counts_Microcystis_16S_counts[,Files[REV_16S_counter]]
  #Normalize by reference length (used average length of Microcystis mcy gene cluster and V4 in the reference database)
  Percent_Toxic_Microcystis[row_tracker, 5] <- as.numeric(Percent_Toxic_Microcystis[row_tracker, 3]) / 11706
  Percent_Toxic_Microcystis[row_tracker, 6] <- as.numeric(Percent_Toxic_Microcystis[row_tracker, 4]) / 293
  #Calculate percent toxic Microcystis
  Percent_Toxic_Microcystis[row_tracker, 7] <- as.numeric(Percent_Toxic_Microcystis[row_tracker, 5]) / as.numeric(Percent_Toxic_Microcystis[row_tracker, 6]) * 100
  
  #Generate data for the all ambig counts:
  row_tracker <- row_tracker + 1
  #Sample ID
  Percent_Toxic_Microcystis[row_tracker, 1] <- Sample
  #Count Method
  Percent_Toxic_Microcystis[row_tracker, 2] <- "All ambig"
  #Total McyD counts
  Percent_Toxic_Microcystis[row_tracker, 3] <- All_ambig_counts_Microcystis_mcyD_counts[,Files[FWD_mcy_counter]] + All_ambig_counts_Microcystis_mcyD_counts[,Files[REV_mcy_counter]]
  #Total 16S V4 counts
  Percent_Toxic_Microcystis[row_tracker, 4] <- All_ambig_counts_Microcystis_16S_counts[,Files[FWD_16S_counter]] + All_ambig_counts_Microcystis_16S_counts[,Files[REV_16S_counter]]
  #Normalize by reference length (used average length of Microcystis mcy gene cluster and V4 in the reference database)
  Percent_Toxic_Microcystis[row_tracker, 5] <- as.numeric(Percent_Toxic_Microcystis[row_tracker, 3]) / 11706
  Percent_Toxic_Microcystis[row_tracker, 6] <- as.numeric(Percent_Toxic_Microcystis[row_tracker, 4]) / 293
  #Calculate percent toxic Microcystis
  Percent_Toxic_Microcystis[row_tracker, 7] <- as.numeric(Percent_Toxic_Microcystis[row_tracker, 5]) / as.numeric(Percent_Toxic_Microcystis[row_tracker, 6]) * 100
  
  #Generate data for the divide ambig counts:
  row_tracker <- row_tracker + 1
  #Sample ID
  Percent_Toxic_Microcystis[row_tracker, 1] <- Sample
  #Count Method
  Percent_Toxic_Microcystis[row_tracker, 2] <- "Divide ambig"
  #Total Mcy counts
  Percent_Toxic_Microcystis[row_tracker, 3] <- Divide_ambig_counts_Microcystis_mcyD_counts[,Files[FWD_mcy_counter]] + Divide_ambig_counts_Microcystis_mcyD_counts[,Files[REV_mcy_counter]]
  #Total 16S V4 counts
  Percent_Toxic_Microcystis[row_tracker, 4] <- Divide_ambig_counts_Microcystis_16S_counts[,Files[FWD_16S_counter]] + Divide_ambig_counts_Microcystis_16S_counts[,Files[REV_16S_counter]]
  #Normalize by reference length (used average length of Microcystis mcy gene cluster and V4 in the reference database)
  Percent_Toxic_Microcystis[row_tracker, 5] <- as.numeric(Percent_Toxic_Microcystis[row_tracker, 3]) / 11706
  Percent_Toxic_Microcystis[row_tracker, 6] <- as.numeric(Percent_Toxic_Microcystis[row_tracker, 4]) / 293
  #Calculate percent toxic Microcystis
  Percent_Toxic_Microcystis[row_tracker, 7] <- as.numeric(Percent_Toxic_Microcystis[row_tracker, 5]) / as.numeric(Percent_Toxic_Microcystis[row_tracker, 6]) * 100
  #Increase counter values for read counts for the next sample:
  FWD_16S_counter <- FWD_16S_counter + 4
  FWD_mcy_counter <- FWD_mcy_counter + 4
  REV_16S_counter <- REV_16S_counter + 4
  REV_mcy_counter <- REV_mcy_counter + 4
  row_tracker <- row_tracker + 1
}

colnames(Percent_Toxic_Microcystis) <- c("sample_ID", "count_method", "mcyD_count", "16S_V4_count", "mcyD_norm_count", "16S_V4_norm_count", "Percent_Toxic")
write.table(Percent_Toxic_Microcystis, file = "Percent_Toxic_Microcystis.txt", quote = FALSE, sep="\t", col.names = TRUE, row.names = FALSE)
```

The below chunk is just to check that the above loop works:  
```{r}
Test <- array(numeric(), c(3,7))
#No ambig, Sample 42895:
Test[1, 1] <- 42896
Test[1, 2] <- "No ambig"
Test[1, 3] <- No_ambig_counts_Microcystis_mcyD_counts$Sample_42896_adtrim_clean_qtrim_fwd_vs_mcy.blastn.postblast.aln_len_filter + No_ambig_counts_Microcystis_mcyD_counts$Sample_42896_adtrim_clean_qtrim_rev_vs_mcy.blastn.postblast.aln_len_filter
Test[1, 4] <- No_ambig_counts_Microcystis_16S_counts$Sample_42896_adtrim_clean_qtrim_fwd_vs_16S.blastn.postblast.aln_len_filter + No_ambig_counts_Microcystis_16S_counts$Sample_42896_adtrim_clean_qtrim_rev_vs_16S.blastn.postblast.aln_len_filter
Test[1, 5] <- as.numeric(Test[1, 3]) / 11706
Test[1, 6] <- as.numeric(Test[1, 4]) / 293
Test[1, 7] <- as.numeric(Test[1, 5]) / as.numeric(Test[1, 6]) * 100

Test[2, 1] <- 49615
Test[2, 2] <- "No ambig"
Test[2, 3] <- No_ambig_counts_Microcystis_mcyD_counts$Sample_49615_adtrim_clean_qtrim_fwd_vs_mcy.blastn.postblast.aln_len_filter + No_ambig_counts_Microcystis_mcyD_counts$Sample_49615_adtrim_clean_qtrim_rev_vs_mcy.blastn.postblast.aln_len_filter
Test[2, 4] <- No_ambig_counts_Microcystis_16S_counts$Sample_49615_adtrim_clean_qtrim_fwd_vs_16S.blastn.postblast.aln_len_filter + No_ambig_counts_Microcystis_16S_counts$Sample_49615_adtrim_clean_qtrim_rev_vs_16S.blastn.postblast.aln_len_filter
Test[2, 5] <- as.numeric(Test[2, 3]) / 11706
Test[2, 6] <- as.numeric(Test[2, 4]) / 293
Test[2, 7] <- as.numeric(Test[2, 5]) / as.numeric(Test[2, 6]) * 100


Test[3, 1] <- 49622
Test[3, 2] <- "No ambig"
Test[3, 3] <- No_ambig_counts_Microcystis_mcyD_counts$Sample_49622_adtrim_clean_qtrim_fwd_vs_mcy.blastn.postblast.aln_len_filter + No_ambig_counts_Microcystis_mcyD_counts$Sample_49622_adtrim_clean_qtrim_rev_vs_mcy.blastn.postblast.aln_len_filter
Test[3, 4] <- No_ambig_counts_Microcystis_16S_counts$Sample_49622_adtrim_clean_qtrim_fwd_vs_16S.blastn.postblast.aln_len_filter + No_ambig_counts_Microcystis_16S_counts$Sample_49622_adtrim_clean_qtrim_rev_vs_16S.blastn.postblast.aln_len_filter
Test[3, 5] <- as.numeric(Test[3, 3]) / 11706
Test[3, 6] <- as.numeric(Test[3, 4]) / 293
Test[3, 7] <- as.numeric(Test[3, 5]) / as.numeric(Test[3, 6]) * 100
```








